<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>pymatgen.io.vasp package &#8212; pymatgen 4.4.9 documentation</title>
    
    <link rel="stylesheet" href="_static/proBlue.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '4.4.9',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="shortcut icon" href="_static/favicon.ico"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="top" title="pymatgen 4.4.9 documentation" href="index.html" />
 
<script type="text/javascript">
  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-33990148-1']);
  _gaq.push(['_trackPageview']);
</script>

  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">pymatgen 4.4.9 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="pymatgen-io-vasp-package">
<h1>pymatgen.io.vasp package<a class="headerlink" href="#pymatgen-io-vasp-package" title="Permalink to this headline">¶</a></h1>
<div class="section" id="subpackages">
<h2>Subpackages<a class="headerlink" href="#subpackages" title="Permalink to this headline">¶</a></h2>
<div class="toctree-wrapper compound">
</div>
</div>
<div class="section" id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="pymatgen-io-vasp-gwvaspinputsets-module">
<h2>pymatgen.io.vasp.GWvaspinputsets module<a class="headerlink" href="#pymatgen-io-vasp-gwvaspinputsets-module" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="module-pymatgen.io.vasp.inputs">
<span id="pymatgen-io-vasp-inputs-module"></span><h2>pymatgen.io.vasp.inputs module<a class="headerlink" href="#module-pymatgen.io.vasp.inputs" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="pymatgen.io.vasp.inputs.Incar">
<em class="property">class </em><code class="descname">Incar</code><span class="sig-paren">(</span><em>params=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Incar"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Incar" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">dict</span></code>, <code class="xref py py-class docutils literal"><span class="pre">monty.json.MSONable</span></code></p>
<p>INCAR object for reading and writing INCAR files. Essentially consists of
a dictionary with some helper functions</p>
<p>Creates an Incar object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>params</strong> (<em>dict</em>) &#8211; A set of input parameters as a dictionary.</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="pymatgen.io.vasp.inputs.Incar.as_dict">
<code class="descname">as_dict</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Incar.as_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Incar.as_dict" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.inputs.Incar.diff">
<code class="descname">diff</code><span class="sig-paren">(</span><em>other</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Incar.diff"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Incar.diff" title="Permalink to this definition">¶</a></dt>
<dd><p>Diff function for Incar.  Compares two Incars and indicates which
parameters are the same and which are not. Useful for checking whether
two runs were done using the same parameters.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>other</strong> (<a class="reference internal" href="#pymatgen.io.vasp.inputs.Incar" title="pymatgen.io.vasp.inputs.Incar"><em>Incar</em></a>) &#8211; The other Incar object to compare to.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">{&#8220;Same&#8221; : parameters_that_are_the_same,
&#8220;Different&#8221;: parameters_that_are_different}
Note that the parameters are return as full dictionaries of values.
E.g. {&#8220;ISIF&#8221;:3}</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Dict of the following format</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="classmethod">
<dt id="pymatgen.io.vasp.inputs.Incar.from_dict">
<em class="property">classmethod </em><code class="descname">from_dict</code><span class="sig-paren">(</span><em>d</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Incar.from_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Incar.from_dict" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.Incar.from_file">
<em class="property">static </em><code class="descname">from_file</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Incar.from_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Incar.from_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads an Incar object from a file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>filename</strong> (<em>str</em>) &#8211; Filename for file</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Incar object</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.Incar.from_string">
<em class="property">static </em><code class="descname">from_string</code><span class="sig-paren">(</span><em>string</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Incar.from_string"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Incar.from_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads an Incar object from a string.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>string</strong> (<em>str</em>) &#8211; Incar string</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Incar object</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.inputs.Incar.get_string">
<code class="descname">get_string</code><span class="sig-paren">(</span><em>sort_keys=False</em>, <em>pretty=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Incar.get_string"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Incar.get_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a string representation of the INCAR.  The reason why this
method is different from the __str__ method is to provide options for
pretty printing.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>sort_keys</strong> (<em>bool</em>) &#8211; Set to True to sort the INCAR parameters
alphabetically. Defaults to False.</li>
<li><strong>pretty</strong> (<em>bool</em>) &#8211; Set to True for pretty aligned output. Defaults
to False.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.Incar.proc_val">
<em class="property">static </em><code class="descname">proc_val</code><span class="sig-paren">(</span><em>key</em>, <em>val</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Incar.proc_val"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Incar.proc_val" title="Permalink to this definition">¶</a></dt>
<dd><p>Static helper method to convert INCAR parameters to proper types, e.g.,
integers, floats, lists, etc.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>key</strong> &#8211; INCAR parameter key</li>
<li><strong>val</strong> &#8211; Actual value of INCAR parameter.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.inputs.Incar.write_file">
<code class="descname">write_file</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Incar.write_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Incar.write_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Write Incar to a file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>filename</strong> (<em>str</em>) &#8211; filename to write to.</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.inputs.Kpoints">
<em class="property">class </em><code class="descname">Kpoints</code><span class="sig-paren">(</span><em>comment='Default gamma'</em>, <em>num_kpts=0</em>, <em>style=&lt;Kpoints_supported_modes.Gamma: 1&gt;</em>, <em>kpts=((1</em>, <em>1</em>, <em>1)</em>, <em>)</em>, <em>kpts_shift=(0</em>, <em>0</em>, <em>0)</em>, <em>kpts_weights=None</em>, <em>coord_type=None</em>, <em>labels=None</em>, <em>tet_number=0</em>, <em>tet_weight=0</em>, <em>tet_connections=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Kpoints"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">monty.json.MSONable</span></code></p>
<p>KPOINT reader/writer.</p>
<p>Highly flexible constructor for Kpoints object.  The flexibility comes
at the cost of usability and in general, it is recommended that you use
the default constructor only if you know exactly what you are doing and
requires the flexibility.  For most usage cases, the three automatic
schemes can be constructed far more easily using the convenience static
constructors (automatic, gamma_automatic, monkhorst_automatic) and it
is recommended that you use those.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>comment</strong> (<em>str</em>) &#8211; String comment for Kpoints</li>
<li><strong>num_kpts</strong> &#8211; Following VASP method of defining the KPOINTS file, this
parameter is the number of kpoints specified. If set to 0
(or negative), VASP automatically generates the KPOINTS.</li>
<li><strong>style</strong> &#8211; Style for generating KPOINTS.  Use one of the
Kpoints.supported_modes enum types.</li>
<li><strong>kpts</strong> (<em>2D array</em>) &#8211; 2D array of kpoints.  Even when only a single
specification is required, e.g. in the automatic scheme,
the kpts should still be specified as a 2D array. e.g.,
[[20]] or [[2,2,2]].</li>
<li><strong>kpts_shift</strong> (<em>3x1 array</em>) &#8211; Shift for Kpoints.</li>
<li><strong>kpts_weights</strong> &#8211; Optional weights for kpoints.  Weights should be
integers. For explicit kpoints.</li>
<li><strong>coord_type</strong> &#8211; In line-mode, this variable specifies whether the
Kpoints were given in Cartesian or Reciprocal coordinates.</li>
<li><strong>labels</strong> &#8211; In line-mode, this should provide a list of labels for
each kpt. It is optional in explicit kpoint mode as comments for
k-points.</li>
<li><strong>tet_number</strong> &#8211; For explicit kpoints, specifies the number of
tetrahedrons for the tetrahedron method.</li>
<li><strong>tet_weight</strong> &#8211; For explicit kpoints, specifies the weight for each
tetrahedron for the tetrahedron method.</li>
<li><strong>tet_connections</strong> &#8211; For explicit kpoints, specifies the connections
of the tetrahedrons for the tetrahedron method.
Format is a list of tuples, [ (sym_weight, [tet_vertices]),
...]</li>
</ul>
</td>
</tr>
</tbody>
</table>
<p>The default behavior of the constructor is for a Gamma centered,
1x1x1 KPOINTS with no shift.</p>
<dl class="method">
<dt id="pymatgen.io.vasp.inputs.Kpoints.as_dict">
<code class="descname">as_dict</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Kpoints.as_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints.as_dict" title="Permalink to this definition">¶</a></dt>
<dd><p>json friendly dict representation of Kpoints</p>
</dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.Kpoints.automatic">
<em class="property">static </em><code class="descname">automatic</code><span class="sig-paren">(</span><em>subdivisions</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Kpoints.automatic"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints.automatic" title="Permalink to this definition">¶</a></dt>
<dd><p>Convenient static constructor for a fully automatic Kpoint grid, with
gamma centered Monkhorst-Pack grids and the number of subdivisions
along each reciprocal lattice vector determined by the scheme in the
VASP manual.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>subdivisions</strong> &#8211; Parameter determining number of subdivisions along
each reciprocal lattice vector.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Kpoints object</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.Kpoints.automatic_density">
<em class="property">static </em><code class="descname">automatic_density</code><span class="sig-paren">(</span><em>structure</em>, <em>kppa</em>, <em>force_gamma=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Kpoints.automatic_density"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints.automatic_density" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns an automatic Kpoint object based on a structure and a kpoint
density. Uses Gamma centered meshes for hexagonal cells and
Monkhorst-Pack grids otherwise.</p>
<dl class="docutils">
<dt>Algorithm:</dt>
<dd>Uses a simple approach scaling the number of divisions along each
reciprocal lattice vector proportional to its length.</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>structure</strong> (<a class="reference internal" href="pymatgen.core.html#pymatgen.core.structure.Structure" title="pymatgen.core.structure.Structure"><em>Structure</em></a>) &#8211; Input structure</li>
<li><strong>kppa</strong> (<em>int</em>) &#8211; Grid density</li>
<li><strong>force_gamma</strong> (<em>bool</em>) &#8211; Force a gamma centered mesh (default is to
use gamma only for hexagonal cells or odd meshes)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">Kpoints</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.Kpoints.automatic_density_by_vol">
<em class="property">static </em><code class="descname">automatic_density_by_vol</code><span class="sig-paren">(</span><em>structure</em>, <em>kppvol</em>, <em>force_gamma=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Kpoints.automatic_density_by_vol"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints.automatic_density_by_vol" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns an automatic Kpoint object based on a structure and a kpoint
density per inverse Angstrom of reciprocal cell.</p>
<dl class="docutils">
<dt>Algorithm:</dt>
<dd>Same as automatic_density()</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>structure</strong> (<a class="reference internal" href="pymatgen.core.html#pymatgen.core.structure.Structure" title="pymatgen.core.structure.Structure"><em>Structure</em></a>) &#8211; Input structure</li>
<li><strong>kppvol</strong> (<em>int</em>) &#8211; Grid density per Angstrom^(-3) of reciprocal cell</li>
<li><strong>force_gamma</strong> (<em>bool</em>) &#8211; Force a gamma centered mesh</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">Kpoints</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.Kpoints.automatic_gamma_density">
<em class="property">static </em><code class="descname">automatic_gamma_density</code><span class="sig-paren">(</span><em>structure</em>, <em>kppa</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Kpoints.automatic_gamma_density"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints.automatic_gamma_density" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns an automatic Kpoint object based on a structure and a kpoint
density. Uses Gamma centered meshes always. For GW.</p>
<dl class="docutils">
<dt>Algorithm:</dt>
<dd>Uses a simple approach scaling the number of divisions along each
reciprocal lattice vector proportional to its length.</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>structure</strong> &#8211; Input structure</li>
<li><strong>kppa</strong> &#8211; Grid density</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.Kpoints.automatic_linemode">
<em class="property">static </em><code class="descname">automatic_linemode</code><span class="sig-paren">(</span><em>divisions</em>, <em>ibz</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Kpoints.automatic_linemode"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints.automatic_linemode" title="Permalink to this definition">¶</a></dt>
<dd><p>Convenient static constructor for a KPOINTS in mode line_mode.
gamma centered Monkhorst-Pack grids and the number of subdivisions
along each reciprocal lattice vector determined by the scheme in the
VASP manual.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>divisions</strong> &#8211; Parameter determining the number of k-points along each
hight symetry lines.</li>
<li><strong>ibz</strong> &#8211; HighSymmKpath object (pymatgen.symmetry.bandstructure)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">Kpoints object</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="classmethod">
<dt id="pymatgen.io.vasp.inputs.Kpoints.from_dict">
<em class="property">classmethod </em><code class="descname">from_dict</code><span class="sig-paren">(</span><em>d</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Kpoints.from_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints.from_dict" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.Kpoints.from_file">
<em class="property">static </em><code class="descname">from_file</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Kpoints.from_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints.from_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads a Kpoints object from a KPOINTS file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>filename</strong> (<em>str</em>) &#8211; filename to read from.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Kpoints object</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.Kpoints.from_string">
<em class="property">static </em><code class="descname">from_string</code><span class="sig-paren">(</span><em>string</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Kpoints.from_string"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints.from_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads a Kpoints object from a KPOINTS string.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>string</strong> (<em>str</em>) &#8211; KPOINTS string.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Kpoints object</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.Kpoints.gamma_automatic">
<em class="property">static </em><code class="descname">gamma_automatic</code><span class="sig-paren">(</span><em>kpts=(1</em>, <em>1</em>, <em>1)</em>, <em>shift=(0</em>, <em>0</em>, <em>0)</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Kpoints.gamma_automatic"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints.gamma_automatic" title="Permalink to this definition">¶</a></dt>
<dd><p>Convenient static constructor for an automatic Gamma centered Kpoint
grid.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>kpts</strong> &#8211; Subdivisions N_1, N_2 and N_3 along reciprocal lattice
vectors. Defaults to (1,1,1)</li>
<li><strong>shift</strong> &#8211; Shift to be applied to the kpoints. Defaults to (0,0,0).</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">Kpoints object</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.Kpoints.monkhorst_automatic">
<em class="property">static </em><code class="descname">monkhorst_automatic</code><span class="sig-paren">(</span><em>kpts=(2</em>, <em>2</em>, <em>2)</em>, <em>shift=(0</em>, <em>0</em>, <em>0)</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Kpoints.monkhorst_automatic"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints.monkhorst_automatic" title="Permalink to this definition">¶</a></dt>
<dd><p>Convenient static constructor for an automatic Monkhorst pack Kpoint
grid.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>kpts</strong> &#8211; Subdivisions N_1, N_2 and N_3 along reciprocal lattice
vectors. Defaults to (2,2,2)</li>
<li><strong>shift</strong> &#8211; Shift to be applied to the kpoints. Defaults to (0,0,0).</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">Kpoints object</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Kpoints.style">
<code class="descname">style</code><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints.style" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Kpoints.supported_modes">
<code class="descname">supported_modes</code><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints.supported_modes" title="Permalink to this definition">¶</a></dt>
<dd><p>alias of <a class="reference internal" href="#pymatgen.io.vasp.inputs.Kpoints_supported_modes" title="pymatgen.io.vasp.inputs.Kpoints_supported_modes"><code class="xref py py-class docutils literal"><span class="pre">Kpoints_supported_modes</span></code></a></p>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.inputs.Kpoints.write_file">
<code class="descname">write_file</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Kpoints.write_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints.write_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Write Kpoints to a file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>filename</strong> (<em>str</em>) &#8211; Filename to write to.</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.inputs.Kpoints_supported_modes">
<em class="property">class </em><code class="descname">Kpoints_supported_modes</code><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Kpoints_supported_modes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints_supported_modes" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">enum.Enum</span></code></p>
<p>An enumeration.</p>
<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Kpoints_supported_modes.Automatic">
<code class="descname">Automatic</code><em class="property"> = &lt;Kpoints_supported_modes.Automatic: 0&gt;</em><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints_supported_modes.Automatic" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Kpoints_supported_modes.Cartesian">
<code class="descname">Cartesian</code><em class="property"> = &lt;Kpoints_supported_modes.Cartesian: 4&gt;</em><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints_supported_modes.Cartesian" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Kpoints_supported_modes.Gamma">
<code class="descname">Gamma</code><em class="property"> = &lt;Kpoints_supported_modes.Gamma: 1&gt;</em><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints_supported_modes.Gamma" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Kpoints_supported_modes.Line_mode">
<code class="descname">Line_mode</code><em class="property"> = &lt;Kpoints_supported_modes.Line_mode: 3&gt;</em><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints_supported_modes.Line_mode" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Kpoints_supported_modes.Monkhorst">
<code class="descname">Monkhorst</code><em class="property"> = &lt;Kpoints_supported_modes.Monkhorst: 2&gt;</em><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints_supported_modes.Monkhorst" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Kpoints_supported_modes.Reciprocal">
<code class="descname">Reciprocal</code><em class="property"> = &lt;Kpoints_supported_modes.Reciprocal: 5&gt;</em><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints_supported_modes.Reciprocal" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.Kpoints_supported_modes.from_string">
<em class="property">static </em><code class="descname">from_string</code><span class="sig-paren">(</span><em>s</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Kpoints_supported_modes.from_string"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Kpoints_supported_modes.from_string" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.inputs.Poscar">
<em class="property">class </em><code class="descname">Poscar</code><span class="sig-paren">(</span><em>structure</em>, <em>comment=None</em>, <em>selective_dynamics=None</em>, <em>true_names=True</em>, <em>velocities=None</em>, <em>predictor_corrector=None</em>, <em>predictor_corrector_preamble=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Poscar"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">monty.json.MSONable</span></code></p>
<p>Object for representing the data in a POSCAR or CONTCAR file.
Please note that this current implementation. Most attributes can be set
directly.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>structure</strong> (<a class="reference internal" href="pymatgen.core.html#pymatgen.core.structure.Structure" title="pymatgen.core.structure.Structure"><em>Structure</em></a>) &#8211; Structure object.</li>
<li><strong>comment</strong> (<em>str</em>) &#8211; Optional comment line for POSCAR. Defaults to unit
cell formula of structure. Defaults to None.</li>
<li><strong>selective_dynamics</strong> (<em>Nx3 array</em>) &#8211; bool values for selective dynamics,
where N is number of sites. Defaults to None.</li>
<li><strong>true_names</strong> (<em>bool</em>) &#8211; Set to False is the names in the POSCAR are not
well-defined and ambiguous. This situation arises commonly in
vasp &lt; 5 where the POSCAR sometimes does not contain element
symbols. Defaults to True.</li>
<li><strong>velocities</strong> (<em>Nx3 array</em>) &#8211; Velocities for the POSCAR. Typically parsed
in MD runs or can be used to initialize velocities.</li>
<li><strong>predictor_corrector</strong> (<em>Nx3 array</em>) &#8211; Predictor corrector for the POSCAR.
Typically parsed in MD runs.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Poscar.structure">
<code class="descname">structure</code><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.structure" title="Permalink to this definition">¶</a></dt>
<dd><p>Associated Structure.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Poscar.comment">
<code class="descname">comment</code><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.comment" title="Permalink to this definition">¶</a></dt>
<dd><p>Optional comment string.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Poscar.true_names">
<code class="descname">true_names</code><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.true_names" title="Permalink to this definition">¶</a></dt>
<dd><p>Boolean indication whether Poscar contains actual real names parsed
from either a POTCAR or the POSCAR itself.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Poscar.selective_dynamics">
<code class="descname">selective_dynamics</code><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.selective_dynamics" title="Permalink to this definition">¶</a></dt>
<dd><p>Selective dynamics attribute for each site if available. A Nx3 array of
booleans.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Poscar.velocities">
<code class="descname">velocities</code><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.velocities" title="Permalink to this definition">¶</a></dt>
<dd><p>Velocities for each site (typically read in from a CONTCAR). A Nx3
array of floats.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Poscar.predictor_corrector">
<code class="descname">predictor_corrector</code><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.predictor_corrector" title="Permalink to this definition">¶</a></dt>
<dd><p>Predictor corrector coordinates and derivatives for each site; i.e.
a list of three 1x3 arrays for each site (typically read in from a MD CONTCAR).</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Poscar.predictor_corrector_preamble">
<code class="descname">predictor_corrector_preamble</code><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.predictor_corrector_preamble" title="Permalink to this definition">¶</a></dt>
<dd><p>Predictor corrector preamble contains the predictor-corrector key, POTIM,
and thermostat parameters that precede the site-specic predictor corrector
data in MD CONTCAR</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Poscar.temperature">
<code class="descname">temperature</code><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.temperature" title="Permalink to this definition">¶</a></dt>
<dd><p>Temperature of velocity Maxwell-Boltzmann initialization. Initialized
to -1 (MB hasn&#8221;t been performed).</p>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.inputs.Poscar.as_dict">
<code class="descname">as_dict</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Poscar.as_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.as_dict" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="classmethod">
<dt id="pymatgen.io.vasp.inputs.Poscar.from_dict">
<em class="property">classmethod </em><code class="descname">from_dict</code><span class="sig-paren">(</span><em>d</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Poscar.from_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.from_dict" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.Poscar.from_file">
<em class="property">static </em><code class="descname">from_file</code><span class="sig-paren">(</span><em>filename</em>, <em>check_for_POTCAR=True</em>, <em>read_velocities=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Poscar.from_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.from_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads a Poscar from a file.</p>
<p>The code will try its best to determine the elements in the POSCAR in
the following order:
1. If check_for_POTCAR is True, the code will try to check if a POTCAR
is in the same directory as the POSCAR and use elements from that by
default. (This is the VASP default sequence of priority).
2. If the input file is Vasp5-like and contains element symbols in the
6th line, the code will use that if check_for_POTCAR is False or there
is no POTCAR found.
3. Failing (2), the code will check if a symbol is provided at the end
of each coordinate.</p>
<p>If all else fails, the code will just assign the first n elements in
increasing atomic number, where n is the number of species, to the
Poscar. For example, H, He, Li, ....  This will ensure at least a
unique element is assigned to each site and any analysis that does not
require specific elemental properties should work fine.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>filename</strong> (<em>str</em>) &#8211; File name containing Poscar data.</li>
<li><strong>check_for_POTCAR</strong> (<em>bool</em>) &#8211; Whether to check if a POTCAR is present
in the same directory as the POSCAR. Defaults to True.</li>
<li><strong>read_velocities</strong> (<em>bool</em>) &#8211; Whether to read or not velocities if they
are present in the POSCAR. Default is True.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">Poscar object.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.Poscar.from_string">
<em class="property">static </em><code class="descname">from_string</code><span class="sig-paren">(</span><em>data</em>, <em>default_names=None</em>, <em>read_velocities=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Poscar.from_string"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.from_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads a Poscar from a string.</p>
<p>The code will try its best to determine the elements in the POSCAR in
the following order:
1. If default_names are supplied and valid, it will use those. Usually,
default names comes from an external source, such as a POTCAR in the
same directory.
2. If there are no valid default names but the input file is Vasp5-like
and contains element symbols in the 6th line, the code will use that.
3. Failing (2), the code will check if a symbol is provided at the end
of each coordinate.</p>
<p>If all else fails, the code will just assign the first n elements in
increasing atomic number, where n is the number of species, to the
Poscar. For example, H, He, Li, ....  This will ensure at least a
unique element is assigned to each site and any analysis that does not
require specific elemental properties should work fine.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>data</strong> (<em>str</em>) &#8211; String containing Poscar data.</li>
<li><strong>default_names</strong> (<em>[str]</em>) &#8211; Default symbols for the POSCAR file,
usually coming from a POTCAR in the same directory.</li>
<li><strong>read_velocities</strong> (<em>bool</em>) &#8211; Whether to read or not velocities if they
are present in the POSCAR. Default is True.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">Poscar object.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.inputs.Poscar.get_string">
<code class="descname">get_string</code><span class="sig-paren">(</span><em>direct=True</em>, <em>vasp4_compatible=False</em>, <em>significant_figures=6</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Poscar.get_string"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.get_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a string to be written as a POSCAR file. By default, site
symbols are written, which means compatibility is for vasp &gt;= 5.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>direct</strong> (<em>bool</em>) &#8211; Whether coordinates are output in direct or
cartesian. Defaults to True.</li>
<li><strong>vasp4_compatible</strong> (<em>bool</em>) &#8211; Set to True to omit site symbols on 6th
line to maintain backward vasp 4.x compatibility. Defaults
to False.</li>
<li><strong>significant_figures</strong> (<em>int</em>) &#8211; No. of significant figures to
output all quantities. Defaults to 6. Note that positions are
output in fixed point, while velocities are output in
scientific format.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">String representation of POSCAR.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Poscar.natoms">
<code class="descname">natoms</code><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.natoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Sequence of number of sites of each type associated with the Poscar.
Similar to 7th line in vasp 5+ POSCAR or the 6th line in vasp 4 POSCAR.</p>
</dd></dl>

<dl class="attribute">
<dt>
<code class="descname">predictor_corrector</code></dt>
<dd></dd></dl>

<dl class="attribute">
<dt>
<code class="descname">selective_dynamics</code></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.inputs.Poscar.set_temperature">
<code class="descname">set_temperature</code><span class="sig-paren">(</span><em>temperature</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Poscar.set_temperature"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.set_temperature" title="Permalink to this definition">¶</a></dt>
<dd><p>Initializes the velocities based on Maxwell-Boltzmann distribution.
Removes linear, but not angular drift (same as VASP)</p>
<p>Scales the energies to the exact temperature (microcanonical ensemble)
Velocities are given in A/fs. This is the vasp default when
direct/cartesian is not specified (even when positions are given in
direct coordinates)</p>
<p>Overwrites imported velocities, if any.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>temperature</strong> (<em>float</em>) &#8211; Temperature in Kelvin.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Poscar.site_symbols">
<code class="descname">site_symbols</code><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.site_symbols" title="Permalink to this definition">¶</a></dt>
<dd><p>Sequence of symbols associated with the Poscar. Similar to 6th line in
vasp 5+ POSCAR.</p>
</dd></dl>

<dl class="attribute">
<dt>
<code class="descname">velocities</code></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.inputs.Poscar.write_file">
<code class="descname">write_file</code><span class="sig-paren">(</span><em>filename</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Poscar.write_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Poscar.write_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Writes POSCAR to a file. The supported kwargs are the same as those for
the Poscar.get_string method and are passed through directly.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.inputs.Potcar">
<em class="property">class </em><code class="descname">Potcar</code><span class="sig-paren">(</span><em>symbols=None</em>, <em>functional='PBE'</em>, <em>sym_potcar_map=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Potcar"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Potcar" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">list</span></code>, <code class="xref py py-class docutils literal"><span class="pre">monty.json.MSONable</span></code></p>
<p>Object for reading and writing POTCAR files for calculations. Consists of a
list of PotcarSingle.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>symbols</strong> (<em>[str]</em>) &#8211; Element symbols for POTCAR. This should correspond
to the symbols used by VASP. E.g., &#8220;Mg&#8221;, &#8220;Fe_pv&#8221;, etc.</li>
<li><strong>functional</strong> (<em>str</em>) &#8211; Functional used. To know what functional options
there are, use Potcar.FUNCTIONAL_CHOICES. Note that VASP has
different versions of the same functional. By default, the old
PBE functional is used. If you want the newer ones, use PBE_52 or
PBE_54. Note that if you intend to compare your results with the
Materials Project, you should use the default setting.</li>
<li><strong>sym_potcar_map</strong> (<em>dict</em>) &#8211; Allows a user to specify a specific element
symbol to raw POTCAR mapping.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Potcar.DEFAULT_FUNCTIONAL">
<code class="descname">DEFAULT_FUNCTIONAL</code><em class="property"> = 'PBE'</em><a class="headerlink" href="#pymatgen.io.vasp.inputs.Potcar.DEFAULT_FUNCTIONAL" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Potcar.FUNCTIONAL_CHOICES">
<code class="descname">FUNCTIONAL_CHOICES</code><em class="property"> = ['LDA', 'PW91', 'LDA_US', 'LDA_54', 'LDA_52', 'PBE_54', 'PBE', 'PBE_52', 'PW91_US']</em><a class="headerlink" href="#pymatgen.io.vasp.inputs.Potcar.FUNCTIONAL_CHOICES" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.inputs.Potcar.as_dict">
<code class="descname">as_dict</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Potcar.as_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Potcar.as_dict" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="classmethod">
<dt id="pymatgen.io.vasp.inputs.Potcar.from_dict">
<em class="property">classmethod </em><code class="descname">from_dict</code><span class="sig-paren">(</span><em>d</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Potcar.from_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Potcar.from_dict" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.Potcar.from_file">
<em class="property">static </em><code class="descname">from_file</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Potcar.from_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Potcar.from_file" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.inputs.Potcar.set_symbols">
<code class="descname">set_symbols</code><span class="sig-paren">(</span><em>symbols</em>, <em>functional='PBE'</em>, <em>sym_potcar_map=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Potcar.set_symbols"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Potcar.set_symbols" title="Permalink to this definition">¶</a></dt>
<dd><p>Initialize the POTCAR from a set of symbols. Currently, the POTCARs can
be fetched from a location specified in .pmgrc.yaml. Use pmg config
to add this setting.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>symbols</strong> (<em>[str]</em>) &#8211; A list of element symbols</li>
<li><strong>functional</strong> (<em>str</em>) &#8211; The functional to use from the config file</li>
<li><strong>sym_potcar_map</strong> (<em>dict</em>) &#8211; A map of symbol:raw POTCAR string. If
sym_potcar_map is specified, POTCARs will be generated from
the given map data rather than the config file location.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Potcar.spec">
<code class="descname">spec</code><a class="headerlink" href="#pymatgen.io.vasp.inputs.Potcar.spec" title="Permalink to this definition">¶</a></dt>
<dd><p>Get the atomic symbols and hash of all the atoms in the POTCAR file.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.Potcar.symbols">
<code class="descname">symbols</code><a class="headerlink" href="#pymatgen.io.vasp.inputs.Potcar.symbols" title="Permalink to this definition">¶</a></dt>
<dd><p>Get the atomic symbols of all the atoms in the POTCAR file.</p>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.inputs.Potcar.write_file">
<code class="descname">write_file</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#Potcar.write_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.Potcar.write_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Write Potcar to a file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>filename</strong> (<em>str</em>) &#8211; filename to write to.</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.inputs.PotcarSingle">
<em class="property">class </em><code class="descname">PotcarSingle</code><span class="sig-paren">(</span><em>*args</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#PotcarSingle"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.PotcarSingle" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#pymatgen.io.vasp.inputs.PotcarSingle" title="pymatgen.io.vasp.inputs.PotcarSingle"><code class="xref py py-class docutils literal"><span class="pre">pymatgen.io.vasp.inputs.PotcarSingle</span></code></a></p>
<p>Object for a <strong>single</strong> POTCAR. The builder assumes the complete string is
the POTCAR contains the complete untouched data in &#8220;data&#8221; as a string and
a dict of keywords.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>data</strong> &#8211; Complete and single potcar file as a string.</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.PotcarSingle.data">
<code class="descname">data</code><a class="headerlink" href="#pymatgen.io.vasp.inputs.PotcarSingle.data" title="Permalink to this definition">¶</a></dt>
<dd><p>POTCAR data as a string.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.inputs.PotcarSingle.keywords">
<code class="descname">keywords</code><a class="headerlink" href="#pymatgen.io.vasp.inputs.PotcarSingle.keywords" title="Permalink to this definition">¶</a></dt>
<dd><p>Keywords parsed from the POTCAR as a dict. All keywords are also
accessible as attributes in themselves. E.g., potcar.enmax,
potcar.encut, etc.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.inputs.VaspInput">
<em class="property">class </em><code class="descname">VaspInput</code><span class="sig-paren">(</span><em>incar</em>, <em>kpoints</em>, <em>poscar</em>, <em>potcar</em>, <em>optional_files=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#VaspInput"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.VaspInput" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">dict</span></code>, <code class="xref py py-class docutils literal"><span class="pre">monty.json.MSONable</span></code></p>
<p>Class to contain a set of vasp input objects corresponding to a run.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>incar</strong> &#8211; Incar object.</li>
<li><strong>kpoints</strong> &#8211; Kpoints object.</li>
<li><strong>poscar</strong> &#8211; Poscar object.</li>
<li><strong>potcar</strong> &#8211; Potcar object.</li>
<li><strong>optional_files</strong> &#8211; Other input files supplied as a dict of {
filename: object}. The object should follow standard pymatgen
conventions in implementing a as_dict() and from_dict method.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="pymatgen.io.vasp.inputs.VaspInput.as_dict">
<code class="descname">as_dict</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#VaspInput.as_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.VaspInput.as_dict" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="classmethod">
<dt id="pymatgen.io.vasp.inputs.VaspInput.from_dict">
<em class="property">classmethod </em><code class="descname">from_dict</code><span class="sig-paren">(</span><em>d</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#VaspInput.from_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.VaspInput.from_dict" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.inputs.VaspInput.from_directory">
<em class="property">static </em><code class="descname">from_directory</code><span class="sig-paren">(</span><em>input_dir</em>, <em>optional_files=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#VaspInput.from_directory"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.VaspInput.from_directory" title="Permalink to this definition">¶</a></dt>
<dd><p>Read in a set of VASP input from a directory. Note that only the
standard INCAR, POSCAR, POTCAR and KPOINTS files are read unless
optional_filenames is specified.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>input_dir</strong> (<em>str</em>) &#8211; Directory to read VASP input from.</li>
<li><strong>optional_files</strong> (<em>dict</em>) &#8211; Optional files to read in as well as a
dict of {filename: Object type}. Object type must have a
static method from_file.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.inputs.VaspInput.write_input">
<code class="descname">write_input</code><span class="sig-paren">(</span><em>output_dir='.'</em>, <em>make_dir_if_not_present=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#VaspInput.write_input"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.VaspInput.write_input" title="Permalink to this definition">¶</a></dt>
<dd><p>Write VASP input to a directory.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>output_dir</strong> (<em>str</em>) &#8211; Directory to write to. Defaults to current
directory (&#8221;.&#8221;).</li>
<li><strong>make_dir_if_not_present</strong> (<em>bool</em>) &#8211; Create the directory if not
present. Defaults to True.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="pymatgen.io.vasp.inputs.parse_bool">
<code class="descname">parse_bool</code><span class="sig-paren">(</span><em>s</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#parse_bool"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.parse_bool" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="pymatgen.io.vasp.inputs.parse_float">
<code class="descname">parse_float</code><span class="sig-paren">(</span><em>s</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#parse_float"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.parse_float" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="pymatgen.io.vasp.inputs.parse_int">
<code class="descname">parse_int</code><span class="sig-paren">(</span><em>s</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#parse_int"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.parse_int" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="pymatgen.io.vasp.inputs.parse_list">
<code class="descname">parse_list</code><span class="sig-paren">(</span><em>s</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#parse_list"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.parse_list" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="pymatgen.io.vasp.inputs.parse_string">
<code class="descname">parse_string</code><span class="sig-paren">(</span><em>s</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/inputs.html#parse_string"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.inputs.parse_string" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</div>
<div class="section" id="module-pymatgen.io.vasp.outputs">
<span id="pymatgen-io-vasp-outputs-module"></span><h2>pymatgen.io.vasp.outputs module<a class="headerlink" href="#module-pymatgen.io.vasp.outputs" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="pymatgen.io.vasp.outputs.BSVasprun">
<em class="property">class </em><code class="descname">BSVasprun</code><span class="sig-paren">(</span><em>filename</em>, <em>parse_projected_eigen=False</em>, <em>parse_potcar_file=False</em>, <em>occu_tol=1e-08</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#BSVasprun"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.BSVasprun" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#pymatgen.io.vasp.outputs.Vasprun" title="pymatgen.io.vasp.outputs.Vasprun"><code class="xref py py-class docutils literal"><span class="pre">pymatgen.io.vasp.outputs.Vasprun</span></code></a></p>
<p>A highly optimized version of Vasprun that parses only eigenvalues for
bandstructures. All other properties like structures, parameters,
etc. are ignored.</p>
<dl class="method">
<dt id="pymatgen.io.vasp.outputs.BSVasprun.as_dict">
<code class="descname">as_dict</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#BSVasprun.as_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.BSVasprun.as_dict" title="Permalink to this definition">¶</a></dt>
<dd><p>Json-serializable dict representation.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.outputs.Chgcar">
<em class="property">class </em><code class="descname">Chgcar</code><span class="sig-paren">(</span><em>poscar</em>, <em>data</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Chgcar"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Chgcar" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#pymatgen.io.vasp.outputs.VolumetricData" title="pymatgen.io.vasp.outputs.VolumetricData"><code class="xref py py-class docutils literal"><span class="pre">pymatgen.io.vasp.outputs.VolumetricData</span></code></a></p>
<p>Simple object for reading a CHGCAR file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>poscar</strong> (<a class="reference internal" href="#pymatgen.io.vasp.inputs.Poscar" title="pymatgen.io.vasp.inputs.Poscar"><em>Poscar</em></a>) &#8211; Poscar object containing structure.</li>
<li><strong>data</strong> &#8211; Actual data.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="staticmethod">
<dt id="pymatgen.io.vasp.outputs.Chgcar.from_file">
<em class="property">static </em><code class="descname">from_file</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Chgcar.from_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Chgcar.from_file" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.outputs.Dynmat">
<em class="property">class </em><code class="descname">Dynmat</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Dynmat"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Dynmat" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">object</span></code></p>
<p>Object for reading a DYNMAT file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>filename</strong> &#8211; Name of file containing DYNMAT.</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Dynmat.data">
<code class="descname">data</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Dynmat.data" title="Permalink to this definition">¶</a></dt>
<dd><p>A nested dict containing the DYNMAT data of the form::
[atom &lt;int&gt;][disp &lt;int&gt;][&#8216;dispvec&#8217;] =</p>
<blockquote>
<div>displacement vector (part of first line in dynmat block, e.g. &#8220;0.01 0 0&#8221;)</div></blockquote>
<dl class="docutils">
<dt>[atom &lt;int&gt;][disp &lt;int&gt;][&#8216;dynmat&#8217;] =</dt>
<dd>&lt;list&gt; list of dynmat lines for this atom and this displacement</dd>
</dl>
</dd></dl>

<p>Authors: Patrick Huck</p>
<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Dynmat.get_phonon_frequencies">
<code class="descname">get_phonon_frequencies</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Dynmat.get_phonon_frequencies"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Dynmat.get_phonon_frequencies" title="Permalink to this definition">¶</a></dt>
<dd><p>calculate phonon frequencies</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Dynmat.masses">
<code class="descname">masses</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Dynmat.masses" title="Permalink to this definition">¶</a></dt>
<dd><p>returns the list of atomic masses</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Dynmat.natoms">
<code class="descname">natoms</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Dynmat.natoms" title="Permalink to this definition">¶</a></dt>
<dd><p>returns the number of atoms</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Dynmat.ndisps">
<code class="descname">ndisps</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Dynmat.ndisps" title="Permalink to this definition">¶</a></dt>
<dd><p>returns the number of displacements</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Dynmat.nspecs">
<code class="descname">nspecs</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Dynmat.nspecs" title="Permalink to this definition">¶</a></dt>
<dd><p>returns the number of species</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.outputs.Locpot">
<em class="property">class </em><code class="descname">Locpot</code><span class="sig-paren">(</span><em>poscar</em>, <em>data</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Locpot"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Locpot" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#pymatgen.io.vasp.outputs.VolumetricData" title="pymatgen.io.vasp.outputs.VolumetricData"><code class="xref py py-class docutils literal"><span class="pre">pymatgen.io.vasp.outputs.VolumetricData</span></code></a></p>
<p>Simple object for reading a LOCPOT file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>poscar</strong> (<a class="reference internal" href="#pymatgen.io.vasp.inputs.Poscar" title="pymatgen.io.vasp.inputs.Poscar"><em>Poscar</em></a>) &#8211; Poscar object containing structure.</li>
<li><strong>data</strong> &#8211; Actual data.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="staticmethod">
<dt id="pymatgen.io.vasp.outputs.Locpot.from_file">
<em class="property">static </em><code class="descname">from_file</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Locpot.from_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Locpot.from_file" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.outputs.Oszicar">
<em class="property">class </em><code class="descname">Oszicar</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Oszicar"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Oszicar" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">object</span></code></p>
<p>A basic parser for an OSZICAR output from VASP.  In general, while the
OSZICAR is useful for a quick look at the output from a VASP run, we
recommend that you use the Vasprun parser instead, which gives far richer
information about a run.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>filename</strong> (<em>str</em>) &#8211; Filename of file to parse</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Oszicar.electronic_steps">
<code class="descname">electronic_steps</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Oszicar.electronic_steps" title="Permalink to this definition">¶</a></dt>
<dd><p>All electronic steps as a list of list of dict. e.g.,
[[{&#8220;rms&#8221;: 160.0, &#8220;E&#8221;: 4507.24605593, &#8220;dE&#8221;: 4507.2, &#8220;N&#8221;: 1,
&#8220;deps&#8221;: -17777.0, &#8220;ncg&#8221;: 16576}, ...], [....]
where electronic_steps[index] refers the list of electronic steps
in one ionic_step, electronic_steps[index][subindex] refers to a
particular electronic step at subindex in ionic step at index. The
dict of properties depends on the type of VASP run, but in general,
&#8220;E&#8221;, &#8220;dE&#8221; and &#8220;rms&#8221; should be present in almost all runs.</p>
</dd></dl>

<dl class="attribute">
<dt>
<code class="descname">ionic_steps:</code></dt>
<dd><p>All ionic_steps as a list of dict, e.g.,
[{&#8220;dE&#8221;: -526.36, &#8220;E0&#8221;: -526.36024, &#8220;mag&#8221;: 0.0, &#8220;F&#8221;: -526.36024},
...]
This is the typical output from VASP at the end of each ionic step.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Oszicar.all_energies">
<code class="descname">all_energies</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Oszicar.all_energies" title="Permalink to this definition">¶</a></dt>
<dd><p>Compilation of all energies from all electronic steps and ionic steps
as a tuple of list of energies, e.g.,
((4507.24605593, 143.824705755, -512.073149912, ...), ...)</p>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Oszicar.as_dict">
<code class="descname">as_dict</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Oszicar.as_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Oszicar.as_dict" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Oszicar.final_energy">
<code class="descname">final_energy</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Oszicar.final_energy" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.outputs.Outcar">
<em class="property">class </em><code class="descname">Outcar</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Outcar"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">monty.json.MSONable</span></code></p>
<p>Parser for data in OUTCAR that is not available in Vasprun.xml</p>
<p>Note, this class works a bit differently than most of the other
VaspObjects, since the OUTCAR can be very different depending on which
&#8220;type of run&#8221; performed.</p>
<p>Creating the OUTCAR class with a filename reads &#8220;regular parameters&#8221; that
are always present.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>filename</strong> (<em>str</em>) &#8211; OUTCAR filename to parse.</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Outcar.magnetization">
<code class="descname">magnetization</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.magnetization" title="Permalink to this definition">¶</a></dt>
<dd><p>Magnetization on each ion as a tuple of dict, e.g.,
({&#8220;d&#8221;: 0.0, &#8220;p&#8221;: 0.003, &#8220;s&#8221;: 0.002, &#8220;tot&#8221;: 0.005}, ... )
Note that this data is not always present.  LORBIT must be set to some
other value than the default.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Outcar.chemical_shifts">
<code class="descname">chemical_shifts</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.chemical_shifts" title="Permalink to this definition">¶</a></dt>
<dd><p>Chemical Shift on each ion as a tuple of ChemicalShiftNotation, e.g.,
(cs1, cs2, ...)</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Outcar.efg">
<code class="descname">efg</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.efg" title="Permalink to this definition">¶</a></dt>
<dd><p>Electric Field Gradient (EFG) tensor on each ion as a tuple of dict, e.g.,
({&#8220;cq&#8221;: 0.1, &#8220;eta&#8221;, 0.2, &#8220;nuclear_quadrupole_moment&#8221;: 0.3},</p>
<blockquote>
<div>{&#8220;cq&#8221;: 0.7, &#8220;eta&#8221;, 0.8, &#8220;nuclear_quadrupole_moment&#8221;: 0.9},
...)</div></blockquote>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Outcar.charge">
<code class="descname">charge</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.charge" title="Permalink to this definition">¶</a></dt>
<dd><p>Charge on each ion as a tuple of dict, e.g.,
({&#8220;p&#8221;: 0.154, &#8220;s&#8221;: 0.078, &#8220;d&#8221;: 0.0, &#8220;tot&#8221;: 0.232}, ...)
Note that this data is not always present.  LORBIT must be set to some
other value than the default.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Outcar.is_stopped">
<code class="descname">is_stopped</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.is_stopped" title="Permalink to this definition">¶</a></dt>
<dd><p>True if OUTCAR is from a stopped run (using STOPCAR, see Vasp Manual).</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Outcar.run_stats">
<code class="descname">run_stats</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.run_stats" title="Permalink to this definition">¶</a></dt>
<dd><p>Various useful run stats as a dict including &#8220;System time (sec)&#8221;,
&#8220;Total CPU time used (sec)&#8221;, &#8220;Elapsed time (sec)&#8221;,
&#8220;Maximum memory used (kb)&#8221;, &#8220;Average memory used (kb)&#8221;,
&#8220;User time (sec)&#8221;.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Outcar.elastic_tensor">
<code class="descname">elastic_tensor</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.elastic_tensor" title="Permalink to this definition">¶</a></dt>
<dt>
<code class="descname">Total elastic moduli (Kbar) is given in a 6x6 array matrix.</code></dt>
<dd></dd></dl>

<p>One can then call a specific reader depending on the type of run being
performed. These are currently: read_igpar(), read_lepsilon() and
read_lcalcpol(), read_core_state_eign().</p>
<p>See the documentation of those methods for more documentation.</p>
<p>Authors: Rickard Armiento, Shyue Ping Ong</p>
<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Outcar.as_dict">
<code class="descname">as_dict</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Outcar.as_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.as_dict" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Outcar.read_chemical_shifts">
<code class="descname">read_chemical_shifts</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Outcar.read_chemical_shifts"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.read_chemical_shifts" title="Permalink to this definition">¶</a></dt>
<dd><p>Parse the NMR chemical shifts data. Only the second part &#8220;absolute, valence and core&#8221;
will be parsed. And only the three right most field (ISO_SHIFT, SPAN, SKEW) will be retrieved.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">List of chemical shifts in the order of atoms from the OUTCAR. Maryland notation is adopted.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Outcar.read_core_state_eigen">
<code class="descname">read_core_state_eigen</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Outcar.read_core_state_eigen"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.read_core_state_eigen" title="Permalink to this definition">¶</a></dt>
<dd><p>Read the core state eigenenergies at each ionic step.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">[core state eig]}].
The core state eigenenergie list for each AO is over all ionic
step.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">A list of dict over the atom such as [{&#8220;AO&#8221;</td>
</tr>
</tbody>
</table>
<p class="rubric">Example</p>
<p>The core state eigenenergie of the 2s AO of the 6th atom of the
structure at the last ionic step is [5][&#8220;2s&#8221;][-1]</p>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Outcar.read_corrections">
<code class="descname">read_corrections</code><span class="sig-paren">(</span><em>reverse=True</em>, <em>terminate_on_match=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Outcar.read_corrections"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.read_corrections" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Outcar.read_elastic_tensor">
<code class="descname">read_elastic_tensor</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Outcar.read_elastic_tensor"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.read_elastic_tensor" title="Permalink to this definition">¶</a></dt>
<dd><p>Parse the elastic tensor data.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">6x6 array corresponding to the elastic tensor from the OUTCAR.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Outcar.read_igpar">
<code class="descname">read_igpar</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Outcar.read_igpar"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.read_igpar" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Renders accessible:</dt>
<dd>er_ev = e&lt;r&gt;_ev (dictionary with Spin.up/Spin.down as keys)
er_bp = e&lt;r&gt;_bp (dictionary with Spin.up/Spin.down as keys)
er_ev_tot = spin up + spin down summed
er_bp_tot = spin up + spin down summed
p_elc = spin up + spin down summed
p_ion = spin up + spin down summed</dd>
</dl>
<p>(See VASP section &#8220;LBERRY,  IGPAR,  NPPSTR,  DIPOL tags&#8221; for info on
what these are).</p>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Outcar.read_lcalcpol">
<code class="descname">read_lcalcpol</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Outcar.read_lcalcpol"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.read_lcalcpol" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Outcar.read_lepsilon">
<code class="descname">read_lepsilon</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Outcar.read_lepsilon"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.read_lepsilon" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Outcar.read_lepsilon_ionic">
<code class="descname">read_lepsilon_ionic</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Outcar.read_lepsilon_ionic"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.read_lepsilon_ionic" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Outcar.read_neb">
<code class="descname">read_neb</code><span class="sig-paren">(</span><em>reverse=True</em>, <em>terminate_on_match=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Outcar.read_neb"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.read_neb" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads NEB data. This only works with OUTCARs from both normal
VASP NEB calculations or from the CI NEB method implemented by
Henkelman et al.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>reverse</strong> (<em>bool</em>) &#8211; Read files in reverse. Defaults to false. Useful for
large files, esp OUTCARs, especially when used with
terminate_on_match. Defaults to True here since we usually
want only the final value.</li>
<li><strong>terminate_on_match</strong> (<em>bool</em>) &#8211; Whether to terminate when there is at
least one match in each key in pattern. Defaults to True here
since we usually want only the final value.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="docutils">
<dt>Renders accessible:</dt>
<dd>tangent_force - Final tangent force.
energy - Final energy.
These can be accessed under Outcar.data[key]</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Outcar.read_nmr_efg">
<code class="descname">read_nmr_efg</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Outcar.read_nmr_efg"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.read_nmr_efg" title="Permalink to this definition">¶</a></dt>
<dd><p>Parse the NMR Electric Field Gradient tensors.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">Electric Field Gradient tensors as a list of dict in the order of atoms from OUTCAR.
Each dict key/value pair corresponds to a component of the tensors.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Outcar.read_pattern">
<code class="descname">read_pattern</code><span class="sig-paren">(</span><em>patterns</em>, <em>reverse=False</em>, <em>terminate_on_match=False</em>, <em>postprocess=&lt;class 'str'&gt;</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Outcar.read_pattern"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.read_pattern" title="Permalink to this definition">¶</a></dt>
<dd><p>General pattern reading. Uses monty&#8217;s regrep method. Takes the same
arguments.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>patterns</strong> (<em>dict</em>) &#8211; A dict of patterns, e.g.,
{&#8220;energy&#8221;: &#8220;energy(sigma-&gt;0)s+=s+([d-.]+)&#8221;}.</li>
<li><strong>reverse</strong> (<em>bool</em>) &#8211; Read files in reverse. Defaults to false. Useful for
large files, esp OUTCARs, especially when used with
terminate_on_match.</li>
<li><strong>terminate_on_match</strong> (<em>bool</em>) &#8211; Whether to terminate when there is at
least one match in each key in pattern.</li>
<li><strong>postprocess</strong> (<em>callable</em>) &#8211; A post processing function to convert all
matches. Defaults to str, i.e., no change.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="docutils">
<dt>Renders accessible:</dt>
<dd>Any attribute in patterns. For example,
{&#8220;energy&#8221;: &#8220;energy(sigma-&gt;0)s+=s+([d-.]+)&#8221;} will set the
value of self.data[&#8220;energy&#8221;] = [[-1234], [-3453], ...], to the
results from regex and postprocess. Note that the returned values
are lists of lists, because you can grep multiple items on one line.</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Outcar.read_piezo_tensor">
<code class="descname">read_piezo_tensor</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Outcar.read_piezo_tensor"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.read_piezo_tensor" title="Permalink to this definition">¶</a></dt>
<dd><p>Parse the piezo tensor data</p>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Outcar.read_table_pattern">
<code class="descname">read_table_pattern</code><span class="sig-paren">(</span><em>header_pattern</em>, <em>row_pattern</em>, <em>footer_pattern</em>, <em>postprocess=&lt;class 'str'&gt;</em>, <em>attribute_name=None</em>, <em>last_one_only=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Outcar.read_table_pattern"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Outcar.read_table_pattern" title="Permalink to this definition">¶</a></dt>
<dd><blockquote>
<div><p>Parse table-like data. A table composes of three parts: header, main body, footer.
All the data matches &#8220;row pattern&#8221; in the main body will be returned.</p>
<dl class="docutils">
<dt>Args:</dt>
<dd><dl class="first last docutils">
<dt>header_pattern (str): The regular expression pattern matches the table header.</dt>
<dd>This pattern should match all the text immediately before the main body of
the table. For multiple sections table match the text until the section of
interest. MULTILINE and DOTALL options are enforced, as a result, the &#8221;.&#8221;
meta-character will also match &#8220;</dd>
</dl>
</dd>
</dl>
</div></blockquote>
<dl class="docutils">
<dt>&#8221; in this section.</dt>
<dd><blockquote class="first">
<div><dl class="docutils">
<dt>row_pattern (str): The regular expression matches a single line in the table.</dt>
<dd>Capture interested field using regular expression groups</dd>
<dt>footer_pattern (str): The regular expression matches the end of the table.</dt>
<dd>E.g. a long dash line.</dd>
<dt>postprocess (callable): A post processing function to convert all</dt>
<dd>matches. Defaults to str, i.e., no change.</dd>
<dt>attribute_name (str): Name of this table. If presense the parsed data will be</dt>
<dd>attached to &#8220;data. e.g. self.data[&#8220;efg&#8221;] = [...]</dd>
<dt>last_one_only (bool): All the tables will be parsed, if this option is set to</dt>
<dd>True, only the last table will be returned. The enclosing list will be removed.
i.e. Only a single table wil be returned. Default to be True.</dd>
</dl>
</div></blockquote>
<dl class="last docutils">
<dt>Returns:</dt>
<dd>List of tables. 1) A table is a list of rows. 2) A row if either a list of
attribute values in case the the capturing group is defined without name in
row_pattern, or a dict in case that named capturing groups are defined by
row_pattern.</dd>
</dl>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.outputs.Procar">
<em class="property">class </em><code class="descname">Procar</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Procar"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Procar" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">object</span></code></p>
<p>Object for reading a PROCAR file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>filename</strong> &#8211; Name of file containing PROCAR.</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Procar.data">
<code class="descname">data</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Procar.data" title="Permalink to this definition">¶</a></dt>
<dd><p>The PROCAR data of the form below. It should VASP uses 1-based indexing,
but all indices are converted to 0-based here.:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="p">{</span>
    <span class="n">spin</span><span class="p">:</span> <span class="n">nd</span><span class="o">.</span><span class="n">array</span> <span class="n">accessed</span> <span class="k">with</span> <span class="p">(</span><span class="n">k</span><span class="o">-</span><span class="n">point</span> <span class="n">index</span><span class="p">,</span> <span class="n">band</span> <span class="n">index</span><span class="p">,</span> <span class="n">ion</span> <span class="n">index</span><span class="p">,</span> <span class="n">orbital</span> <span class="n">index</span><span class="p">)</span>
<span class="p">}</span>
</pre></div>
</div>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Procar.weights">
<code class="descname">weights</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Procar.weights" title="Permalink to this definition">¶</a></dt>
<dd><p>The weights associated with each k-point as an nd.array of lenght
nkpoints.</p>
</dd></dl>

<p>..attribute:: phase_factors</p>
<blockquote>
<div><p>Phase factors, where present (e.g. LORBIT = 12). A dict of the form:
{</p>
<blockquote>
<div>spin: complex nd.array accessed with (k-point index, band index, ion index, orbital index)</div></blockquote>
<p>}</p>
</div></blockquote>
<p>..attribute:: nbands</p>
<blockquote>
<div>Number of bands</div></blockquote>
<p>..attribute:: nkpoints</p>
<blockquote>
<div>Number of k-points</div></blockquote>
<p>..attribute:: nions</p>
<blockquote>
<div>Number of ions</div></blockquote>
<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Procar.get_occupation">
<code class="descname">get_occupation</code><span class="sig-paren">(</span><em>atom_index</em>, <em>orbital</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Procar.get_occupation"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Procar.get_occupation" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the occupation for a particular orbital of a particular atom.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>atom_num</strong> (<em>int</em>) &#8211; Index of atom in the PROCAR. It should be noted
that VASP uses 1-based indexing for atoms, but this is
converted to 0-based indexing in this parser to be
consistent with representation of structures in pymatgen.</li>
<li><strong>orbital</strong> (<em>str</em>) &#8211; An orbital. If it is a single character, e.g., s,
p, d or f, the sum of all s-type, p-type, d-type or f-type
orbitals occupations are returned respectively. If it is a
specific orbital, e.g., px, dxy, etc., only the occupation
of that orbital is returned.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">Sum occupation of orbital of atom.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Procar.get_projection_on_elements">
<code class="descname">get_projection_on_elements</code><span class="sig-paren">(</span><em>structure</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Procar.get_projection_on_elements"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Procar.get_projection_on_elements" title="Permalink to this definition">¶</a></dt>
<dd><p>Method returning a dictionary of projections on elements.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>structure</strong> (<a class="reference internal" href="pymatgen.core.html#pymatgen.core.structure.Structure" title="pymatgen.core.structure.Structure"><em>Structure</em></a>) &#8211; Input structure.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">[k index][b index][{Element:values}]]</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">a dictionary in the {Spin.up</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="exception">
<dt id="pymatgen.io.vasp.outputs.UnconvergedVASPWarning">
<em class="property">exception </em><code class="descname">UnconvergedVASPWarning</code><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#UnconvergedVASPWarning"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.UnconvergedVASPWarning" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">Warning</span></code></p>
<p>Warning for unconverged vasp run.</p>
</dd></dl>

<dl class="exception">
<dt id="pymatgen.io.vasp.outputs.VaspParserError">
<em class="property">exception </em><code class="descname">VaspParserError</code><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#VaspParserError"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.VaspParserError" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">Exception</span></code></p>
<p>Exception class for VASP parsing.</p>
</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.outputs.Vasprun">
<em class="property">class </em><code class="descname">Vasprun</code><span class="sig-paren">(</span><em>filename</em>, <em>ionic_step_skip=None</em>, <em>ionic_step_offset=0</em>, <em>parse_dos=True</em>, <em>parse_eigen=True</em>, <em>parse_projected_eigen=False</em>, <em>parse_potcar_file=True</em>, <em>occu_tol=1e-08</em>, <em>exception_on_bad_xml=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Vasprun"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">monty.json.MSONable</span></code></p>
<p>Vastly improved cElementTree-based parser for vasprun.xml files. Uses
iterparse to support incremental parsing of large files.
Speedup over Dom is at least 2x for smallish files (~1Mb) to orders of
magnitude for larger files (~10Mb).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>filename</strong> (<em>str</em>) &#8211; Filename to parse</li>
<li><strong>ionic_step_skip</strong> (<em>int</em>) &#8211; If ionic_step_skip is a number &gt; 1,
only every ionic_step_skip ionic steps will be read for
structure and energies. This is very useful if you are parsing
very large vasprun.xml files and you are not interested in every
single ionic step. Note that the final energies may not be the
actual final energy in the vasprun.</li>
<li><strong>ionic_step_offset</strong> (<em>int</em>) &#8211; Used together with ionic_step_skip. If set,
the first ionic step read will be offset by the amount of
ionic_step_offset. For example, if you want to start reading
every 10th structure but only from the 3rd structure onwards,
set ionic_step_skip to 10 and ionic_step_offset to 3. Main use
case is when doing statistical structure analysis with
extremely long time scale multiple VASP calculations of
varying numbers of steps.</li>
<li><strong>parse_dos</strong> (<em>bool</em>) &#8211; Whether to parse the dos. Defaults to True. Set
to False to shave off significant time from the parsing if you
are not interested in getting those data.</li>
<li><strong>parse_eigen</strong> (<em>bool</em>) &#8211; Whether to parse the eigenvalues. Defaults to
True. Set to False to shave off significant time from the
parsing if you are not interested in getting those data.</li>
<li><strong>parse_projected_eigen</strong> (<em>bool</em>) &#8211; Whether to parse the projected
eigenvalues. Defaults to False. Set to True to obtain projected
eigenvalues. <strong>Note that this can take an extreme amount of time
and memory.</strong> So use this wisely.</li>
<li><strong>parse_potcar_file</strong> (<em>bool/str</em>) &#8211; <p>Whether to parse the potcar file to read
the potcar hashes for the potcar_spec attribute. Defaults to True,
where no hashes will be determined and the potcar_spec dictionaries
will read {&#8220;symbol&#8221;: ElSymbol, &#8220;hash&#8221;: None}. By Default, looks in
the same directory as the vasprun.xml, with same extensions as</p>
<blockquote>
<div>Vasprun.xml. If a string is provided, looks at that filepath.</div></blockquote>
</li>
<li><strong>occu_tol</strong> (<em>float</em>) &#8211; Sets the minimum tol for the determination of the
vbm and cbm. Usually the default of 1e-8 works well enough,
but there may be pathological cases.</li>
<li><strong>exception_on_bad_xml</strong> (<em>bool</em>) &#8211; Whether to throw a ParseException if a
malformed XML is detected. Default to True, which ensures only
proper vasprun.xml are parsed. You can set to False if you want
partial results (e.g., if you are monitoring a calculation during a
run), but use the results with care. A warning is issued.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<p><strong>Vasp results</strong></p>
<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.ionic_steps">
<code class="descname">ionic_steps</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.ionic_steps" title="Permalink to this definition">¶</a></dt>
<dd><p>All ionic steps in the run as a list of
{&#8220;structure&#8221;: structure at end of run,
&#8220;electronic_steps&#8221;: {All electronic step data in vasprun file},
&#8220;stresses&#8221;: stress matrix}</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.structures">
<code class="descname">structures</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.structures" title="Permalink to this definition">¶</a></dt>
<dd><p>List of Structure objects for the structure at each ionic step.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.tdos">
<code class="descname">tdos</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.tdos" title="Permalink to this definition">¶</a></dt>
<dd><p>Total dos calculated at the end of run.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.idos">
<code class="descname">idos</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.idos" title="Permalink to this definition">¶</a></dt>
<dd><p>Integrated dos calculated at the end of run.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.pdos">
<code class="descname">pdos</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.pdos" title="Permalink to this definition">¶</a></dt>
<dd><p>List of list of PDos objects. Access as pdos[atomindex][orbitalindex]</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.efermi">
<code class="descname">efermi</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.efermi" title="Permalink to this definition">¶</a></dt>
<dd><p>Fermi energy</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.eigenvalues">
<code class="descname">eigenvalues</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.eigenvalues" title="Permalink to this definition">¶</a></dt>
<dd><p>Available only if parse_eigen=True. Final eigenvalues as a dict of
{(spin, kpoint index):[[eigenvalue, occu]]}.
This representation is based on actual ordering in VASP and is meant as
an intermediate representation to be converted into proper objects. The
kpoint index is 0-based (unlike the 1-based indexing in VASP).</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.projected_eigenvalues">
<code class="descname">projected_eigenvalues</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.projected_eigenvalues" title="Permalink to this definition">¶</a></dt>
<dd><p>Final projected eigenvalues as a dict of {spin: nd-array}. To access
a particular value, you need to do
Vasprun.projected_eigenvalues[spin][kpoint index][band index][atom index][orbital_index]
This representation is based on actual ordering in VASP and is meant as
an intermediate representation to be converted into proper objects. The
kpoint, band and atom indices are 0-based (unlike the 1-based indexing
in VASP).</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.dielectric">
<code class="descname">dielectric</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.dielectric" title="Permalink to this definition">¶</a></dt>
<dd><p>The real and imaginary part of the dielectric constant (e.g., computed
by RPA) in function of the energy (frequency). Optical properties (e.g.
absorption coefficient) can be obtained through this.
The data is given as a tuple of 3 values containing each of them
the energy, the real part tensor, and the imaginary part tensor
([energies],[[real_partxx,real_partyy,real_partzz,real_partxy,
real_partyz,real_partxz]],[[imag_partxx,imag_partyy,imag_partzz,
imag_partxy, imag_partyz, imag_partxz]])</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.other_dielectric">
<code class="descname">other_dielectric</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.other_dielectric" title="Permalink to this definition">¶</a></dt>
<dd><p>Dictionary, with the tag comment as key, containing other variants of
the real and imaginary part of the dielectric constant (e.g., computed
by RPA) in function of the energy (frequency). Optical properties (e.g.
absorption coefficient) can be obtained through this.
The data is given as a tuple of 3 values containing each of them
the energy, the real part tensor, and the imaginary part tensor
([energies],[[real_partxx,real_partyy,real_partzz,real_partxy,
real_partyz,real_partxz]],[[imag_partxx,imag_partyy,imag_partzz,
imag_partxy, imag_partyz, imag_partxz]])</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.epsilon_static">
<code class="descname">epsilon_static</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.epsilon_static" title="Permalink to this definition">¶</a></dt>
<dd><p>The static part of the dielectric constant. Present when it&#8217;s a DFPT run
(LEPSILON=TRUE)</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.epsilon_static_wolfe">
<code class="descname">epsilon_static_wolfe</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.epsilon_static_wolfe" title="Permalink to this definition">¶</a></dt>
<dd><p>The static part of the dielectric constant without any local field
effects. Present when it&#8217;s a DFPT run (LEPSILON=TRUE)</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.epsilon_ionic">
<code class="descname">epsilon_ionic</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.epsilon_ionic" title="Permalink to this definition">¶</a></dt>
<dd><p>The ionic part of the static dielectric constant. Present when it&#8217;s a
DFPT run (LEPSILON=TRUE) and IBRION=5, 6, 7 or 8</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.nionic_steps">
<code class="descname">nionic_steps</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.nionic_steps" title="Permalink to this definition">¶</a></dt>
<dd><p>The total number of ionic steps. This number is always equal
to the total number of steps in the actual run even if
ionic_step_skip is used.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.force_constants">
<code class="descname">force_constants</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.force_constants" title="Permalink to this definition">¶</a></dt>
<dd><p>Force constants computed in phonon DFPT run(IBRION = 8).
The data is a 4D numpy array of shape (natoms, natoms, 3, 3).</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.normalmode_eigenvals">
<code class="descname">normalmode_eigenvals</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.normalmode_eigenvals" title="Permalink to this definition">¶</a></dt>
<dd><p>Normal mode frequencies.
1D numpy array of size 3*natoms.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.normalmode_eigenvecs">
<code class="descname">normalmode_eigenvecs</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.normalmode_eigenvecs" title="Permalink to this definition">¶</a></dt>
<dd><p>Normal mode eigen vectors.
3D numpy array of shape (3*natoms, natoms, 3).</p>
</dd></dl>

<p><strong>Vasp inputs</strong></p>
<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.incar">
<code class="descname">incar</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.incar" title="Permalink to this definition">¶</a></dt>
<dd><p>Incar object for parameters specified in INCAR file.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.parameters">
<code class="descname">parameters</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.parameters" title="Permalink to this definition">¶</a></dt>
<dd><p>Incar object with parameters that vasp actually used, including all
defaults.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.kpoints">
<code class="descname">kpoints</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.kpoints" title="Permalink to this definition">¶</a></dt>
<dd><p>Kpoints object for KPOINTS specified in run.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.actual_kpoints">
<code class="descname">actual_kpoints</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.actual_kpoints" title="Permalink to this definition">¶</a></dt>
<dd><p>List of actual kpoints, e.g.,
[[0.25, 0.125, 0.08333333], [-0.25, 0.125, 0.08333333],
[0.25, 0.375, 0.08333333], ....]</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.actual_kpoints_weights">
<code class="descname">actual_kpoints_weights</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.actual_kpoints_weights" title="Permalink to this definition">¶</a></dt>
<dd><p>List of kpoint weights, E.g.,
[0.04166667, 0.04166667, 0.04166667, 0.04166667, 0.04166667, ....]</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.atomic_symbols">
<code class="descname">atomic_symbols</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.atomic_symbols" title="Permalink to this definition">¶</a></dt>
<dd><p>List of atomic symbols, e.g., [&#8220;Li&#8221;, &#8220;Fe&#8221;, &#8220;Fe&#8221;, &#8220;P&#8221;, &#8220;P&#8221;, &#8220;P&#8221;]</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.potcar_symbols">
<code class="descname">potcar_symbols</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.potcar_symbols" title="Permalink to this definition">¶</a></dt>
<dd><p>List of POTCAR symbols. e.g.,
[&#8220;PAW_PBE Li 17Jan2003&#8221;, &#8220;PAW_PBE Fe 06Sep2000&#8221;, ..]</p>
</dd></dl>

<p>Author: Shyue Ping Ong</p>
<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Vasprun.as_dict">
<code class="descname">as_dict</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Vasprun.as_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.as_dict" title="Permalink to this definition">¶</a></dt>
<dd><p>Json-serializable dict representation.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.complete_dos">
<code class="descname">complete_dos</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.complete_dos" title="Permalink to this definition">¶</a></dt>
<dd><p>A complete dos object which incorporates the total dos and all
projected dos.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.converged">
<code class="descname">converged</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.converged" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns true if a relaxation run is converged.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.converged_electronic">
<code class="descname">converged_electronic</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.converged_electronic" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks that electronic step convergence has been reached in the final
ionic step</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.converged_ionic">
<code class="descname">converged_ionic</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.converged_ionic" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks that ionic step convergence has been reached, i.e. that vasp
exited before reaching the max ionic steps for a relaxation run</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.eigenvalue_band_properties">
<code class="descname">eigenvalue_band_properties</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.eigenvalue_band_properties" title="Permalink to this definition">¶</a></dt>
<dd><p>Band properties from the eigenvalues as a tuple,
(band gap, cbm, vbm, is_band_gap_direct).</p>
</dd></dl>

<dl class="attribute">
<dt>
<code class="descname">epsilon_ionic</code></dt>
<dd><p>Property only available for DFPT calculations and when IBRION=5, 6, 7 or 8.</p>
</dd></dl>

<dl class="attribute">
<dt>
<code class="descname">epsilon_static</code></dt>
<dd><p>Property only available for DFPT calculations.</p>
</dd></dl>

<dl class="attribute">
<dt>
<code class="descname">epsilon_static_wolfe</code></dt>
<dd><p>Property only available for DFPT calculations.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.final_energy">
<code class="descname">final_energy</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.final_energy" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Vasprun.get_band_structure">
<code class="descname">get_band_structure</code><span class="sig-paren">(</span><em>kpoints_filename=None</em>, <em>efermi=None</em>, <em>line_mode=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Vasprun.get_band_structure"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.get_band_structure" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the band structure as a BandStructure object</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>kpoints_filename</strong> (<em>str</em>) &#8211; Full path of the KPOINTS file from which
the band structure is generated.
If none is provided, the code will try to intelligently
determine the appropriate KPOINTS file by substituting the
filename of the vasprun.xml with KPOINTS.
The latter is the default behavior.</li>
<li><strong>efermi</strong> (<em>float</em>) &#8211; If you want to specify manually the fermi energy
this is where you should do it. By default, the None value
means the code will get it from the vasprun.</li>
<li><strong>line_mode</strong> (<em>bool</em>) &#8211; Force the band structure to be considered as
a run along symmetry lines.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last"><p>a BandStructure object (or more specifically a
BandStructureSymmLine object if the run is detected to be a run
along symmetry lines)</p>
<p>Two types of runs along symmetry lines are accepted: non-sc with
Line-Mode in the KPOINT file or hybrid, self-consistent with a
uniform grid+a few kpoints along symmetry lines (explicit KPOINTS
file) (it&#8217;s not possible to run a non-sc band structure with hybrid
functionals). The explicit KPOINTS file needs to have data on the
kpoint label as commentary.</p>
</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Vasprun.get_computed_entry">
<code class="descname">get_computed_entry</code><span class="sig-paren">(</span><em>inc_structure=False</em>, <em>parameters=None</em>, <em>data=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Vasprun.get_computed_entry"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.get_computed_entry" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a ComputedStructureEntry from the vasprun.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>inc_structure</strong> (<em>bool</em>) &#8211; Set to True if you want
ComputedStructureEntries to be returned instead of
ComputedEntries.</li>
<li><strong>parameters</strong> (<em>list</em>) &#8211; Input parameters to include. It has to be one of
the properties supported by the Vasprun object. If
parameters == None, a default set of parameters that are
necessary for typical post-processing will be set.</li>
<li><strong>data</strong> (<em>list</em>) &#8211; Output data to include. Has to be one of the properties
supported by the Vasprun object.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">ComputedStructureEntry/ComputedEntry</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.hubbards">
<code class="descname">hubbards</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.hubbards" title="Permalink to this definition">¶</a></dt>
<dd><p>Hubbard U values used if a vasprun is a GGA+U run. {} otherwise.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.is_hubbard">
<code class="descname">is_hubbard</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.is_hubbard" title="Permalink to this definition">¶</a></dt>
<dd><p>True if run is a DFT+U run.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.is_spin">
<code class="descname">is_spin</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.is_spin" title="Permalink to this definition">¶</a></dt>
<dd><p>True if run is spin-polarized.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.lattice">
<code class="descname">lattice</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.lattice" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.lattice_rec">
<code class="descname">lattice_rec</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.lattice_rec" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Vasprun.run_type">
<code class="descname">run_type</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.run_type" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the run type. Currently supports only GGA and HF calcs.</p>
<p>TODO: Fix for other functional types like LDA, PW91, etc.</p>
</dd></dl>

<dl class="attribute">
<dt>
<code class="descname">structures</code></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Vasprun.update_potcar_spec">
<code class="descname">update_potcar_spec</code><span class="sig-paren">(</span><em>path</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Vasprun.update_potcar_spec"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Vasprun.update_potcar_spec" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.outputs.VolumetricData">
<em class="property">class </em><code class="descname">VolumetricData</code><span class="sig-paren">(</span><em>structure</em>, <em>data</em>, <em>distance_matrix=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#VolumetricData"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.VolumetricData" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">object</span></code></p>
<p>Simple volumetric object for reading LOCPOT and CHGCAR type files.</p>
<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.VolumetricData.structure">
<code class="descname">structure</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.VolumetricData.structure" title="Permalink to this definition">¶</a></dt>
<dd><p>Structure associated with the Volumetric Data object</p>
</dd></dl>

<p>..attribute:: is_spin_polarized</p>
<blockquote>
<div>True if run is spin polarized</div></blockquote>
<p>..attribute:: dim</p>
<blockquote>
<div>Tuple of dimensions of volumetric grid in each direction (nx, ny, nz).</div></blockquote>
<p>..attribute:: data</p>
<blockquote>
<div>Actual data as a dict of {string: np.array}. The string are &#8220;total&#8221;
and &#8220;diff&#8221;, in accordance to the output format of vasp LOCPOT and
CHGCAR files where the total spin density is written first, followed
by the difference spin density.</div></blockquote>
<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.VolumetricData.ngridpts">
<code class="descname">ngridpts</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.VolumetricData.ngridpts" title="Permalink to this definition">¶</a></dt>
<dd><p>Total number of grid points in volumetric data.</p>
</dd></dl>

<p>Typically, this constructor is not used directly and the static
from_file constructor is used. This constructor is designed to allow
summation and other operations between VolumetricData objects.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>structure</strong> &#8211; Structure associated with the volumetric data</li>
<li><strong>data</strong> &#8211; Actual volumetric data.</li>
<li><strong>distance_matrix</strong> &#8211; A pre-computed distance matrix if available.
Useful so pass distance_matrices between sums,
shortcircuiting an otherwise expensive operation.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="pymatgen.io.vasp.outputs.VolumetricData.get_average_along_axis">
<code class="descname">get_average_along_axis</code><span class="sig-paren">(</span><em>ind</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#VolumetricData.get_average_along_axis"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.VolumetricData.get_average_along_axis" title="Permalink to this definition">¶</a></dt>
<dd><p>Get the averaged total of the volumetric data a certain axis direction.
For example, useful for visualizing Hartree Potentials from a LOCPOT
file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>ind</strong> (<em>int</em>) &#8211; Index of axis.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Average total along axis</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.VolumetricData.get_axis_grid">
<code class="descname">get_axis_grid</code><span class="sig-paren">(</span><em>ind</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#VolumetricData.get_axis_grid"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.VolumetricData.get_axis_grid" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the grid for a particular axis.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>ind</strong> (<em>int</em>) &#8211; Axis index.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.VolumetricData.get_integrated_diff">
<code class="descname">get_integrated_diff</code><span class="sig-paren">(</span><em>ind</em>, <em>radius</em>, <em>nbins=1</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#VolumetricData.get_integrated_diff"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.VolumetricData.get_integrated_diff" title="Permalink to this definition">¶</a></dt>
<dd><p>Get integrated difference of atom index ind up to radius. This can be
an extremely computationally intensive process, depending on how many
grid points are in the VolumetricData.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>ind</strong> (<em>int</em>) &#8211; Index of atom.</li>
<li><strong>radius</strong> (<em>float</em>) &#8211; Radius of integration.</li>
<li><strong>nbins</strong> (<em>int</em>) &#8211; Number of bins. Defaults to 1. This allows one to
obtain the charge integration up to a list of the cumulative
charge integration values for radii for [radius/nbins,
2 * radius/nbins, ....].</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">Differential integrated charge as a np array of [[radius, value],
...]. Format is for ease of plotting. E.g., plt.plot(data[:,0],
data[:,1])</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.VolumetricData.linear_add">
<code class="descname">linear_add</code><span class="sig-paren">(</span><em>other</em>, <em>scale_factor=1.0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#VolumetricData.linear_add"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.VolumetricData.linear_add" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to do a linear sum of volumetric objects. Used by + and -
operators as well. Returns a VolumetricData object containing the
linear sum.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>other</strong> (<a class="reference internal" href="#pymatgen.io.vasp.outputs.VolumetricData" title="pymatgen.io.vasp.outputs.VolumetricData"><em>VolumetricData</em></a>) &#8211; Another VolumetricData object</li>
<li><strong>scale_factor</strong> (<em>float</em>) &#8211; Factor to scale the other data by.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">VolumetricData corresponding to self + scale_factor * other.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="staticmethod">
<dt id="pymatgen.io.vasp.outputs.VolumetricData.parse_file">
<em class="property">static </em><code class="descname">parse_file</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#VolumetricData.parse_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.VolumetricData.parse_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Convenience method to parse a generic volumetric data file in the vasp
like format. Used by subclasses for parsing file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>filename</strong> (<em>str</em>) &#8211; Path of file to parse</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">(poscar, data)</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.VolumetricData.spin_data">
<code class="descname">spin_data</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.VolumetricData.spin_data" title="Permalink to this definition">¶</a></dt>
<dd><p><em>The data decomposed into actual spin data as {spin</em> &#8211; data}.
Essentially, this provides the actual Spin.up and Spin.down data
instead of the total and diff.  Note that by definition, a
non-spin-polarized run would have Spin.up data == Spin.down data.</p>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.VolumetricData.write_file">
<code class="descname">write_file</code><span class="sig-paren">(</span><em>file_name</em>, <em>vasp4_compatible=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#VolumetricData.write_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.VolumetricData.write_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Write the VolumetricData object to a vasp compatible file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>file_name</strong> (<em>str</em>) &#8211; Path to a file</li>
<li><strong>vasp4_compatible</strong> (<em>bool</em>) &#8211; True if the format is vasp4 compatible</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.outputs.Wavederf">
<em class="property">class </em><code class="descname">Wavederf</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Wavederf"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Wavederf" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">object</span></code></p>
<p>Object for reading a WAVEDERF file.</p>
<p>Note: This file is only produced when LOPTICS is true AND vasp has been
recompiled after uncommenting the line that calls
WRT_CDER_BETWEEN_STATES_FORMATTED in linear_optics.F</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>filename</strong> &#8211; Name of file containing WAVEDERF.</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Wavederf.data">
<code class="descname">data</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Wavederf.data" title="Permalink to this definition">¶</a></dt>
<dd><p>A numpy array containing the WAVEDERF data of the form below. It should
be noted that VASP uses 1-based indexing for bands, but this is
converted to 0-based numpy array indexing.</p>
<p>For each kpoint (in the same order as in IBZKPT), and for each pair of
bands:</p>
<blockquote>
<div><dl class="docutils">
<dt>[ #kpoint index</dt>
<dd><dl class="first docutils">
<dt>[ #band 1 index</dt>
<dd><dl class="first docutils">
<dt>[ #band 2 index</dt>
<dd>[cdum_x_real, cdum_x_imag, cdum_y_real, cdum_y_imag, cdum_z_real, cdum_z_imag]</dd>
</dl>
<p class="last">]</p>
</dd>
</dl>
<p class="last">]</p>
</dd>
</dl>
<p>]</p>
</div></blockquote>
<p>This structure follows the file format. Numpy array methods can be used
to fetch data in a more useful way (e.g., get matrix elements between
wo specific bands at each kpoint, fetch x/y/z components,
real/imaginary parts, abs/phase, etc. )</p>
</dd></dl>

<p>Author: Miguel Dias Costa</p>
<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Wavederf.get_elements_between_bands">
<code class="descname">get_elements_between_bands</code><span class="sig-paren">(</span><em>band_i</em>, <em>band_j</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Wavederf.get_elements_between_bands"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Wavederf.get_elements_between_bands" title="Permalink to this definition">¶</a></dt>
<dd><p>Method returning a numpy array with elements</p>
<p>[cdum_x_real, cdum_x_imag, cdum_y_real, cdum_y_imag, cdum_z_real, cdum_z_imag]</p>
<p>between bands band_i and band_j (vasp 1-based indexing) for all kpoints.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>band_i</strong> (<em>Integer</em>) &#8211; Index of band i</li>
<li><strong>band_j</strong> (<em>Integer</em>) &#8211; Index of band j</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">a numpy list of elements for each kpoint</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Wavederf.nb_bands">
<code class="descname">nb_bands</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Wavederf.nb_bands" title="Permalink to this definition">¶</a></dt>
<dd><p>returns the number of bands in the band structure</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Wavederf.nb_kpoints">
<code class="descname">nb_kpoints</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Wavederf.nb_kpoints" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the number of k-points in the band structure calculation</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.outputs.Xdatcar">
<em class="property">class </em><code class="descname">Xdatcar</code><span class="sig-paren">(</span><em>filename</em>, <em>ionicstep_start=1</em>, <em>ionicstep_end=None</em>, <em>comment=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Xdatcar"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Xdatcar" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">object</span></code></p>
<p>Class representing an XDATCAR file. Only tested with VASP 5.x files.</p>
<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Xdatcar.structures">
<code class="descname">structures</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Xdatcar.structures" title="Permalink to this definition">¶</a></dt>
<dd><p>List of structures parsed from XDATCAR.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Xdatcar.comment">
<code class="descname">comment</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Xdatcar.comment" title="Permalink to this definition">¶</a></dt>
<dd><p>Optional comment string.</p>
</dd></dl>

<p>Authors: Ram Balachandran</p>
<p>Init a Xdatcar.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>filename</strong> (<em>str</em>) &#8211; Filename of input XDATCAR file.</li>
<li><strong>ionicstep_start</strong> (<em>int</em>) &#8211; Starting number of ionic step.</li>
<li><strong>ionicstep_end</strong> (<em>int</em>) &#8211; Ending number of ionic step.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Xdatcar.concatenate">
<code class="descname">concatenate</code><span class="sig-paren">(</span><em>filename</em>, <em>ionicstep_start=1</em>, <em>ionicstep_end=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Xdatcar.concatenate"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Xdatcar.concatenate" title="Permalink to this definition">¶</a></dt>
<dd><p>Concatenate structures in file to Xdatcar.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>filename</strong> (<em>str</em>) &#8211; Filename of XDATCAR file to be concatenated.</li>
<li><strong>ionicstep_start</strong> (<em>int</em>) &#8211; Starting number of ionic step.</li>
<li><strong>ionicstep_end</strong> (<em>int</em>) &#8211; Ending number of ionic step.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="docutils">
<dt>TODO(rambalachandran):</dt>
<dd>Requires a check to ensure if the new concatenating file has the same lattice structure and atoms as the Xdatcar class.</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Xdatcar.get_string">
<code class="descname">get_string</code><span class="sig-paren">(</span><em>ionicstep_start=1</em>, <em>ionicstep_end=None</em>, <em>significant_figures=8</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Xdatcar.get_string"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Xdatcar.get_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Write  Xdatcar class into a file
:param filename: Filename of output XDATCAR file.
:type filename: str
:param ionicstep_start: Starting number of ionic step.
:type ionicstep_start: int
:param ionicstep_end: Ending number of ionic step.
:type ionicstep_end: int</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Xdatcar.natoms">
<code class="descname">natoms</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Xdatcar.natoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Sequence of number of sites of each type associated with the Poscar.
Similar to 7th line in vasp 5+ Xdatcar.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.outputs.Xdatcar.site_symbols">
<code class="descname">site_symbols</code><a class="headerlink" href="#pymatgen.io.vasp.outputs.Xdatcar.site_symbols" title="Permalink to this definition">¶</a></dt>
<dd><p>Sequence of symbols associated with the Xdatcar. Similar to 6th line in
vasp 5+ Xdatcar.</p>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.outputs.Xdatcar.write_file">
<code class="descname">write_file</code><span class="sig-paren">(</span><em>filename</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#Xdatcar.write_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.Xdatcar.write_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Write  Xdatcar class into a file.
:param filename: Filename of output XDATCAR file.
:type filename: str
:param The supported kwargs are the same as those for the Xdatcar.get_string method and are passed through directly.:</p>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="pymatgen.io.vasp.outputs.get_adjusted_fermi_level">
<code class="descname">get_adjusted_fermi_level</code><span class="sig-paren">(</span><em>efermi</em>, <em>cbm</em>, <em>band_structure</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#get_adjusted_fermi_level"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.get_adjusted_fermi_level" title="Permalink to this definition">¶</a></dt>
<dd><p>When running a band structure computations the fermi level needs to be
take from the static run that gave the charge density used for the non-self
consistent band structure run. Sometimes this fermi level is however a
little too low because of the mismatch between the uniform grid used in
the static run and the band structure k-points (e.g., the VBM is on Gamma
and the Gamma point is not in the uniform mesh). Here we use a procedure
consisting in looking for energy levels higher than the static fermi level
(but lower than the LUMO) if any of these levels make the band structure
appears insulating and not metallic anymore, we keep this adjusted fermi
level. This procedure has shown to detect correctly most insulators.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>efermi</strong> (<em>float</em>) &#8211; Fermi energy of the static run</li>
<li><strong>cbm</strong> (<em>float</em>) &#8211; Conduction band minimum of the static run</li>
<li><strong>run_bandstructure</strong> &#8211; a band_structure object</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">a new adjusted fermi level</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pymatgen.io.vasp.outputs.get_band_structure_from_vasp_multiple_branches">
<code class="descname">get_band_structure_from_vasp_multiple_branches</code><span class="sig-paren">(</span><em>dir_name</em>, <em>efermi=None</em>, <em>projections=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/outputs.html#get_band_structure_from_vasp_multiple_branches"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.outputs.get_band_structure_from_vasp_multiple_branches" title="Permalink to this definition">¶</a></dt>
<dd><p>This method is used to get band structure info from a VASP directory. It
takes into account that the run can be divided in several branches named
&#8220;branch_x&#8221;. If the run has not been divided in branches the method will
turn to parsing vasprun.xml directly.</p>
<p>The method returns None is there&#8221;s a parsing error</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>dir_name</strong> &#8211; Directory containing all bandstructure runs.</li>
<li><strong>efermi</strong> &#8211; Efermi for bandstructure.</li>
<li><strong>projections</strong> &#8211; True if you want to get the data on site projections if
any. Note that this is sometimes very large</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">A BandStructure Object</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-pymatgen.io.vasp.sets">
<span id="pymatgen-io-vasp-sets-module"></span><h2>pymatgen.io.vasp.sets module<a class="headerlink" href="#module-pymatgen.io.vasp.sets" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="pymatgen.io.vasp.sets.DictSet">
<em class="property">class </em><code class="descname">DictSet</code><span class="sig-paren">(</span><em>structure</em>, <em>config_dict</em>, <em>files_to_transfer=None</em>, <em>user_incar_settings=None</em>, <em>user_kpoints_settings=None</em>, <em>constrain_total_magmom=False</em>, <em>sort_structure=True</em>, <em>potcar_functional='PBE'</em>, <em>force_gamma=False</em>, <em>reduce_structure=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#DictSet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.DictSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#pymatgen.io.vasp.sets.VaspInputSet" title="pymatgen.io.vasp.sets.VaspInputSet"><code class="xref py py-class docutils literal"><span class="pre">pymatgen.io.vasp.sets.VaspInputSet</span></code></a></p>
<p>Concrete implementation of VaspInputSet that is initialized from a dict
settings. This allows arbitrary settings to be input. In general,
this is rarely used directly unless there is a source of settings in yaml
format (e.g., from a REST interface). It is typically used by other
VaspInputSets for initialization.</p>
<p>Special consideration should be paid to the way the MAGMOM initialization
for the INCAR is done. The initialization differs depending on the type of
structure and the configuration settings. The order in which the magmom is
determined is as follows:</p>
<ol class="arabic simple">
<li>If the site itself has a magmom setting, that is used.</li>
<li>If the species on the site has a spin setting, that is used.</li>
<li>If the species itself has a particular setting in the config file, that
is used, e.g., Mn3+ may have a different magmom than Mn4+.</li>
<li>Lastly, the element symbol itself is checked in the config file. If
there are no settings, VASP&#8217;s default of 0.6 is used.</li>
</ol>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>structure</strong> (<a class="reference internal" href="pymatgen.core.html#pymatgen.core.structure.Structure" title="pymatgen.core.structure.Structure"><em>Structure</em></a>) &#8211; The Structure to create inputs for.</li>
<li><strong>config_dict</strong> (<em>dict</em>) &#8211; The config dictionary to use.</li>
<li><strong>files_to_transfer</strong> (<em>dict</em>) &#8211; A dictionary of {filename: filepath}. This
allows the transfer of files from a previous calculation.</li>
<li><strong>user_incar_settings</strong> (<em>dict</em>) &#8211; User INCAR settings. This allows a user
to override INCAR settings, e.g., setting a different MAGMOM for
various elements or species. Note that in the new scheme,
ediff_per_atom and hubbard_u are no longer args. Instead, the
config_dict supports EDIFF_PER_ATOM and EDIFF keys. The former
scales with # of atoms, the latter does not. If both are
present, EDIFF is preferred. To force such settings, just supply
user_incar_settings={&#8220;EDIFF&#8221;: 1e-5, &#8220;LDAU&#8221;: False} for example.</li>
<li><strong>user_kpoints_settings</strong> (<em>dict</em>) &#8211; Allow user to override kpoints setting by
supplying a dict. E.g., {&#8220;reciprocal_density&#8221;: 1000}. Default is
None.</li>
<li><strong>constrain_total_magmom</strong> (<em>bool</em>) &#8211; Whether to constrain the total magmom
(NUPDOWN in INCAR) to be the sum of the expected MAGMOM for all
species. Defaults to False.</li>
<li><strong>sort_structure</strong> (<em>bool</em>) &#8211; Whether to sort the structure (using the
default sort order of electronegativity) before generating input
files. Defaults to True, the behavior you would want most of the
time. This ensures that similar atomic species are grouped
together.</li>
<li><strong>potcar_functional</strong> (<em>str</em>) &#8211; Functional to use. Default (None) is to use
the functional in Potcar.DEFAULT_FUNCTIONAL. Valid values:
&#8220;PBE&#8221;, &#8220;LDA&#8221;, &#8220;PW91&#8221;, &#8220;LDA_US&#8221;</li>
<li><strong>force_gamma</strong> (<em>bool</em>) &#8211; Force gamma centered kpoint generation. Default
(False) is to use the Automatic Density kpoint scheme, which
will use the Gamma centered generation scheme for hexagonal
cells, and Monkhorst-Pack otherwise.</li>
<li><strong>reduce_structure</strong> (<em>None/str</em>) &#8211; Before generating the input files,
generate the reduced structure. Default (None), does not
alter the structure. Valid values: None, &#8220;niggli&#8221;, &#8220;LLL&#8221;.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.DictSet.incar">
<code class="descname">incar</code><a class="headerlink" href="#pymatgen.io.vasp.sets.DictSet.incar" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.DictSet.kpoints">
<code class="descname">kpoints</code><a class="headerlink" href="#pymatgen.io.vasp.sets.DictSet.kpoints" title="Permalink to this definition">¶</a></dt>
<dd><p>Writes out a KPOINTS file using the fully automated grid method. Uses
Gamma centered meshes  for hexagonal cells and Monk grids otherwise.</p>
<dl class="docutils">
<dt>Algorithm:</dt>
<dd>Uses a simple approach scaling the number of divisions along each
reciprocal lattice vector proportional to its length.</dd>
</dl>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.DictSet.nelect">
<code class="descname">nelect</code><a class="headerlink" href="#pymatgen.io.vasp.sets.DictSet.nelect" title="Permalink to this definition">¶</a></dt>
<dd><p>Gets the default number of electrons for a given structure.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.DictSet.poscar">
<code class="descname">poscar</code><a class="headerlink" href="#pymatgen.io.vasp.sets.DictSet.poscar" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.sets.DictSet.write_input">
<code class="descname">write_input</code><span class="sig-paren">(</span><em>output_dir</em>, <em>make_dir_if_not_present=True</em>, <em>include_cif=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#DictSet.write_input"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.DictSet.write_input" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.sets.MITMDSet">
<em class="property">class </em><code class="descname">MITMDSet</code><span class="sig-paren">(</span><em>structure</em>, <em>start_temp</em>, <em>end_temp</em>, <em>nsteps</em>, <em>time_step=2</em>, <em>spin_polarized=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#MITMDSet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.MITMDSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#pymatgen.io.vasp.sets.MITRelaxSet" title="pymatgen.io.vasp.sets.MITRelaxSet"><code class="xref py py-class docutils literal"><span class="pre">pymatgen.io.vasp.sets.MITRelaxSet</span></code></a></p>
<p>Class for writing a vasp md run. This DOES NOT do multiple stage
runs.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>start_temp</strong> (<em>int</em>) &#8211; Starting temperature.</li>
<li><strong>end_temp</strong> (<em>int</em>) &#8211; Final temperature.</li>
<li><strong>nsteps</strong> (<em>int</em>) &#8211; Number of time steps for simulations. The NSW parameter.</li>
<li><strong>time_step</strong> (<em>int</em>) &#8211; The time step for the simulation. The POTIM
parameter. Defaults to 2fs.</li>
<li><strong>spin_polarized</strong> (<em>bool</em>) &#8211; Whether to do spin polarized calculations.
The ISPIN parameter. Defaults to False.</li>
<li><strong>sort_structure</strong> (<em>bool</em>) &#8211; Whether to sort structure. Defaults to False
(different behavior from standard input sets).</li>
<li><strong>**kwargs</strong> &#8211; Other kwargs supported by <a class="reference internal" href="#pymatgen.io.vasp.sets.DictSet" title="pymatgen.io.vasp.sets.DictSet"><code class="xref py py-class docutils literal"><span class="pre">DictSet</span></code></a>.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.MITMDSet.kpoints">
<code class="descname">kpoints</code><a class="headerlink" href="#pymatgen.io.vasp.sets.MITMDSet.kpoints" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.sets.MITNEBSet">
<em class="property">class </em><code class="descname">MITNEBSet</code><span class="sig-paren">(</span><em>structures</em>, <em>unset_encut=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#MITNEBSet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.MITNEBSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#pymatgen.io.vasp.sets.MITRelaxSet" title="pymatgen.io.vasp.sets.MITRelaxSet"><code class="xref py py-class docutils literal"><span class="pre">pymatgen.io.vasp.sets.MITRelaxSet</span></code></a></p>
<p>Class for writing NEB inputs. Note that EDIFF is not on a per atom
basis for this input set.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>unset_encut</strong> (<em>bool</em>) &#8211; Whether to unset ENCUT.</li>
<li><strong>**kwargs</strong> &#8211; Other kwargs supported by <a class="reference internal" href="#pymatgen.io.vasp.sets.DictSet" title="pymatgen.io.vasp.sets.DictSet"><code class="xref py py-class docutils literal"><span class="pre">DictSet</span></code></a>.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.MITNEBSet.poscar">
<code class="descname">poscar</code><a class="headerlink" href="#pymatgen.io.vasp.sets.MITNEBSet.poscar" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.MITNEBSet.poscars">
<code class="descname">poscars</code><a class="headerlink" href="#pymatgen.io.vasp.sets.MITNEBSet.poscars" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.sets.MITNEBSet.write_input">
<code class="descname">write_input</code><span class="sig-paren">(</span><em>output_dir</em>, <em>make_dir_if_not_present=True</em>, <em>write_cif=False</em>, <em>write_path_cif=False</em>, <em>write_endpoint_inputs=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#MITNEBSet.write_input"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.MITNEBSet.write_input" title="Permalink to this definition">¶</a></dt>
<dd><p>NEB inputs has a special directory structure where inputs are in 00,
01, 02, ....</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>output_dir</strong> (<em>str</em>) &#8211; Directory to output the VASP input files</li>
<li><strong>make_dir_if_not_present</strong> (<em>bool</em>) &#8211; Set to True if you want the
directory (and the whole path) to be created if it is not
present.</li>
<li><strong>write_cif</strong> (<em>bool</em>) &#8211; If true, writes a cif along with each POSCAR.</li>
<li><strong>write_path_cif</strong> (<em>bool</em>) &#8211; If true, writes a cif for each image.</li>
<li><strong>write_endpoint_inputs</strong> (<em>bool</em>) &#8211; If true, writes input files for
running endpoint calculations.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.sets.MITRelaxSet">
<em class="property">class </em><code class="descname">MITRelaxSet</code><span class="sig-paren">(</span><em>structure</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#MITRelaxSet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.MITRelaxSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#pymatgen.io.vasp.sets.DictSet" title="pymatgen.io.vasp.sets.DictSet"><code class="xref py py-class docutils literal"><span class="pre">pymatgen.io.vasp.sets.DictSet</span></code></a></p>
<p>Standard implementation of VaspInputSet utilizing parameters in the MIT
High-throughput project.
The parameters are chosen specifically for a high-throughput project,
which means in general pseudopotentials with fewer electrons were chosen.</p>
<p>Please refer:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">A</span> <span class="n">Jain</span><span class="p">,</span> <span class="n">G</span><span class="o">.</span> <span class="n">Hautier</span><span class="p">,</span> <span class="n">C</span><span class="o">.</span> <span class="n">Moore</span><span class="p">,</span> <span class="n">S</span><span class="o">.</span> <span class="n">P</span><span class="o">.</span> <span class="n">Ong</span><span class="p">,</span> <span class="n">C</span><span class="o">.</span> <span class="n">Fischer</span><span class="p">,</span> <span class="n">T</span><span class="o">.</span> <span class="n">Mueller</span><span class="p">,</span>
<span class="n">K</span><span class="o">.</span> <span class="n">A</span><span class="o">.</span> <span class="n">Persson</span><span class="p">,</span> <span class="n">G</span><span class="o">.</span> <span class="n">Ceder</span><span class="o">.</span> <span class="n">A</span> <span class="n">high</span><span class="o">-</span><span class="n">throughput</span> <span class="n">infrastructure</span> <span class="k">for</span> <span class="n">density</span>
<span class="n">functional</span> <span class="n">theory</span> <span class="n">calculations</span><span class="o">.</span> <span class="n">Computational</span> <span class="n">Materials</span> <span class="n">Science</span><span class="p">,</span>
<span class="mi">2011</span><span class="p">,</span> <span class="mi">50</span><span class="p">(</span><span class="mi">8</span><span class="p">),</span> <span class="mi">2295</span><span class="o">-</span><span class="mf">2310.</span> <span class="n">doi</span><span class="p">:</span><span class="mf">10.1016</span><span class="o">/</span><span class="n">j</span><span class="o">.</span><span class="n">commatsci</span><span class="o">.</span><span class="mf">2011.02</span><span class="o">.</span><span class="mi">023</span>
</pre></div>
</div>
<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.MITRelaxSet.CONFIG">
<code class="descname">CONFIG</code><em class="property"> = {'KPOINTS': {'grid_density': 1000}, 'POTCAR': {'Th': {'hash': 'aea79f322180fa6f0bfa74cb2a156dcf', 'symbol': 'Th'}, 'Rb': {'hash': 'e447c648d870b066b3514e6b800727ab', 'symbol': 'Rb_pv'}, 'Ca': {'hash': 'eb006721e214c04b3c13146e81b3a27d', 'symbol': 'Ca_sv'}, 'Tm': {'hash': '94a07cb7949b01305cb161da0cbfb492', 'symbol': 'Tm_3'}, 'Sc': {'hash': 'dc386f505ad0c43385a7715b4111cb75', 'symbol': 'Sc_sv'}, 'Pm': {'hash': 'a2c9485ea86b2a7cf175077e6e5c7b3e', 'symbol': 'Pm'}, 'Si': {'hash': 'b2b0ea6feb62e7cde209616683b8f7f5', 'symbol': 'Si'}, 'Mn': {'hash': 'd082dba29b57ab59b3165e605dbf71b8', 'symbol': 'Mn'}, 'Tl': {'hash': '2aa0d5406aaab7ebfbc761da382f1352', 'symbol': 'Tl'}, 'Os': {'hash': '35c2cb48d48a9c38c40fb82bbe70626d', 'symbol': 'Os'}, 'Rh': {'hash': '2c3dba3fcc6058ca1b1cfa75e45084bc', 'symbol': 'Rh'}, 'Zn': {'hash': 'e35ee27f8483a63bb68dbc236a343af3', 'symbol': 'Zn'}, 'Xe': {'hash': '338472e581f58b41d37c002a5e22353b', 'symbol': 'Xe'}, 'Nb': {'hash': '7bcee99a4dc3094be0f9fd7961c02966', 'symbol': 'Nb_pv'}, 'K': {'hash': '3e84f86d37f203a4fb01de36af57e430', 'symbol': 'K_sv'}, 'Ti': {'hash': 'c617e8b539c3f44a0ab6e8da2a92d318', 'symbol': 'Ti'}, 'O': {'hash': '7a25bc5b9a5393f46600a4939d357982', 'symbol': 'O'}, 'Pt': {'hash': 'a604ea3c6a9cc23c739b762f625cf449', 'symbol': 'Pt'}, 'Er': {'hash': 'daa65a04877317f8c3c593ddeaa8a132', 'symbol': 'Er_3'}, 'Bi': {'hash': 'e29661c79d59abae3b3ba69eae24b1a5', 'symbol': 'Bi'}, 'C': {'hash': 'c0a8167dbb174fe492a3db7f5006c0f8', 'symbol': 'C'}, 'Sb': {'hash': 'd82c022b02fc5344e85bd1909f9ee3e7', 'symbol': 'Sb'}, 'Sr': {'hash': 'ca6a5429c120a0ab705824386a76fe5b', 'symbol': 'Sr_sv'}, 'P': {'hash': '7dc3393307131ae67785a0cdacb61d5f', 'symbol': 'P'}, 'Cd': {'hash': '0506b2d0ac28d5fe2b5ced77a701aa86', 'symbol': 'Cd'}, 'Pa': {'hash': 'a1fdb1089d0727f415416ec8082246ba', 'symbol': 'Pa'}, 'U': {'hash': '72702eabbb1bc02b4167590dc848ed5d', 'symbol': 'U'}, 'Pb': {'hash': '704c2c967247d7f84090d2536c91877d', 'symbol': 'Pb'}, 'Eu': {'hash': 'd466d046adf21f6146ee9644049ea268', 'symbol': 'Eu'}, 'Kr': {'hash': '39b9b85ae3982e6c012fb549b2840ce5', 'symbol': 'Kr'}, 'Li': {'hash': '65e83282d1707ec078c1012afbd05be8', 'symbol': 'Li'}, 'Ga': {'hash': '6e0b9d58412b1bfcd7252aff13d476c2', 'symbol': 'Ga'}, 'Dy': {'hash': 'd4a05220ab0a2d4c03a76872ea724a1e', 'symbol': 'Dy_3'}, 'Ni': {'hash': '653f5772e68b2c7fd87ffd1086c0d710', 'symbol': 'Ni'}, 'Ag': {'hash': 'e8ffa02fe3f3a51338ac1ac91ae968b9', 'symbol': 'Ag'}, 'Ge': {'hash': '79e788788c31e196a460553010512d3f', 'symbol': 'Ge'}, 'H': {'hash': 'bb43c666e3d36577264afe07669e9582', 'symbol': 'H'}, 'Ce': {'hash': 'ff3a09f2ff91798e58eb4b9854e9be4a', 'symbol': 'Ce'}, 'He': {'hash': '47f9434aa3db96c85d7c4b3e4c2df09b', 'symbol': 'He'}, 'Re': {'hash': '72385e193c92a8acfe17ea49004c2be1', 'symbol': 'Re'}, 'V': {'hash': '7f1297a2e1d963e2a4d81b61f85e4ded', 'symbol': 'V_pv'}, 'In': {'hash': '7df38c0cdb4e6d9a9b93f09d690bb3ae', 'symbol': 'In'}, 'Se': {'hash': '67a8804ede9f1112726e3d136978ef19', 'symbol': 'Se'}, 'Ir': {'hash': 'dbcf7dcc6f4fb40df7b3d26904f60a66', 'symbol': 'Ir'}, 'Cl': {'hash': '779b9901046c78fe51c5d80224642aeb', 'symbol': 'Cl'}, 'W': {'hash': '2a33e0d5c700640535f60ac0a12177ab', 'symbol': 'W_pv'}, 'Cu': {'hash': '8ca4e43a30de0c397e51f16bbb20d678', 'symbol': 'Cu'}, 'Ho': {'hash': '661891464a27e87cf7e1324dd1893b77', 'symbol': 'Ho_3'}, 'Ne': {'hash': '52064eee378b9e37a295a674f1c278f0', 'symbol': 'Ne'}, 'Ac': {'hash': 'd6854224d20e3de6e6fd7399503791d1', 'symbol': 'Ac'}, 'Te': {'hash': '72719856e22fb1d3032df6f96d98a0f2', 'symbol': 'Te'}, 'N': {'hash': 'b98fd027ddebc67da4063ff2cabbc04b', 'symbol': 'N'}, 'Nd': {'hash': '0c64e63070cee837c967283fffa001df', 'symbol': 'Nd'}, 'Ta': {'hash': 'd4e2cfe9338ef80da592d5bb9dc782c7', 'symbol': 'Ta'}, 'Gd': {'hash': '1f0d42b1e5f6769d319d3f247992aeb9', 'symbol': 'Gd'}, 'Fe': {'hash': '9530da8244e4dac17580869b4adab115', 'symbol': 'Fe'}, 'Pd': {'hash': 'a395eb3aaf2fcab12fac3030a1146f61', 'symbol': 'Pd'}, 'Np': {'hash': '20cb30b714200c4db870550b288ac4cd', 'symbol': 'Np'}, 'As': {'hash': '8005364db225a254e52cba350bedd032', 'symbol': 'As'}, 'Co': {'hash': 'b169bca4e137294d2ab3df8cbdd09083', 'symbol': 'Co'}, 'Br': {'hash': '40f9594b4506684a69158c8975cfb9d6', 'symbol': 'Br'}, 'F': {'hash': '180141c33d032bfbfff30b3bea9d23dd', 'symbol': 'F'}, 'La': {'hash': '9b3ce03d18f7c0b40471a817ff91b287', 'symbol': 'La'}, 'Be': {'hash': 'fb974e44d56a8c62c6bbd1a1eb70c3a7', 'symbol': 'Be'}, 'Tb': {'hash': '0790955c547003956c0fd4f080f7f508', 'symbol': 'Tb_3'}, 'Na': {'hash': '1a89e79f7e21d99e8cf5788979f6a987', 'symbol': 'Na'}, 'Zr': {'hash': 'd221d2c0bac4f8e81af2f5c42a314274', 'symbol': 'Zr'}, 'Cs': {'hash': '096b53a7d80cc0086976bcda50d536e5', 'symbol': 'Cs_sv'}, 'Sm': {'hash': 'e5e274e7cd99602ca81d146155abdf88', 'symbol': 'Sm_3'}, 'Tc': {'hash': '9592642886319309a39d55c5717c6f48', 'symbol': 'Tc'}, 'Al': {'hash': 'a6fd9a46aec185f4ad2acd0cbe4ae2fa', 'symbol': 'Al'}, 'B': {'hash': '18ed2875dfa6305324cec3d7d59273ae', 'symbol': 'B'}, 'Ar': {'hash': 'e782fc6292623b396091bf8b871c272f', 'symbol': 'Ar'}, 'Hf': {'hash': 'b113f150cbf9c736f8244a6c25b0482e', 'symbol': 'Hf'}, 'Ba': {'hash': 'c0477913afb63dfae3439f3534fbf0ed', 'symbol': 'Ba_sv'}, 'Mg': {'hash': '1771eb72adbbfa6310d66e7517e49930', 'symbol': 'Mg'}, 'Ru': {'hash': '7925f4d4b68076d70af7cd86eef9ba8d', 'symbol': 'Ru_pv'}, 'Pr': {'hash': '92f191499bf5346ea652bb806350ad87', 'symbol': 'Pr'}, 'S': {'hash': 'd368db6899d8839859bbee4811a42a88', 'symbol': 'S'}, 'Y': {'hash': '4ed187e77cd54f198bb88020278b143d', 'symbol': 'Y_sv'}, 'Hg': {'hash': 'c2f15dfb5fd53396c5427635e5019160', 'symbol': 'Hg'}, 'Yb': {'hash': '9f472bd422f640710f7d93e2d9ce89f4', 'symbol': 'Yb'}, 'Lu': {'hash': 'd40a90babf1224b88ffb4c3273ac3848', 'symbol': 'Lu_3'}, 'Sn': {'hash': '849b0795e148f93113a06be8fd5f5001', 'symbol': 'Sn_d'}, 'Mo': {'hash': '84e18fd84a98e3d7fa8f055952410df0', 'symbol': 'Mo_pv'}, 'I': {'hash': 'f4ff16a495dd361ff5824ee61b418bb0', 'symbol': 'I'}, 'Pu': {'hash': 'f1d01e845dccc52d448679911f301a73', 'symbol': 'Pu'}, 'Cr': {'hash': '82c14307937c7509fda4e9bc023d243d', 'symbol': 'Cr'}, 'Au': {'hash': 'a9182d436a13194b744640ac940ab9b0', 'symbol': 'Au'}}, 'INCAR': {'LWAVE': False, 'LDAUL': {'F': {'V': 2, 'Nb': 2, 'Mo': 2, 'Ta': 2, 'Fe': 2, 'W': 2, 'Cu': 2, 'Mn': 2, 'Cr': 2, 'Ag': 2, 'Ni': 2, 'Co': 2, 'Re': 2}, 'O': {'V': 2, 'Nb': 2, 'Mo': 2, 'Ta': 2, 'Fe': 2, 'W': 2, 'Cu': 2, 'Mn': 2, 'Cr': 2, 'Ag': 2, 'Ni': 2, 'Co': 2, 'Re': 2}, 'S': {'Mn': 2.5, 'Fe': 2}}, 'NELM': 100, 'PREC': 'Accurate', 'NSW': 99, 'LDAUTYPE': 2, 'EDIFF_PER_ATOM': 5e-05, 'ENCUT': 520, 'NELMIN': 6, 'LDAUPRINT': 1, 'ISIF': 3, 'LORBIT': '11', 'LDAUU': {'F': {'V': 3.1, 'Nb': 1.5, 'Mo': 4.38, 'Ta': 2, 'Fe': 4.0, 'W': 4.0, 'Cu': 4, 'Mn': 3.9, 'Cr': 3.5, 'Ag': 1.5, 'Ni': 6, 'Co': 3.4, 'Re': 2}, 'O': {'V': 3.1, 'Nb': 1.5, 'Mo': 4.38, 'Ta': 2, 'Fe': 4.0, 'W': 4.0, 'Cu': 4, 'Mn': 3.9, 'Cr': 3.5, 'Ag': 1.5, 'Ni': 6, 'Co': 3.4, 'Re': 2}, 'S': {'Mn': 2.5, 'Fe': 1.9}}, 'ISMEAR': -5, 'ICHARG': 1, 'LDAUJ': {'F': {'V': 0, 'Nb': 0, 'Mo': 0, 'Ta': 0, 'Fe': 0, 'W': 0, 'Cu': 0, 'Mn': 0, 'Cr': 0, 'Ag': 0, 'Ni': 0, 'Co': 0, 'Re': 0}, 'O': {'V': 0, 'Nb': 0, 'Mo': 0, 'Ta': 0, 'Fe': 0, 'W': 0, 'Cu': 0, 'Mn': 0, 'Cr': 0, 'Ag': 0, 'Ni': 0, 'Co': 0, 'Re': 0}, 'S': {'Mn': 0, 'Fe': 0}}, 'SIGMA': 0.05, 'LREAL': 'AUTO', 'ISPIN': 2, 'MAGMOM': {'V': 5, 'Ni3+': 1, 'Co3+': 0.6, 'Ta': 5, 'Fe': 5, 'Co4+': 1, 'Mn': 5, 'Cr': 5, 'Mn3+': 4, 'Co': 5, 'Ni4+': 0.6, 'Mn4+': 3, 'Mn2+': 5, 'Fe4+': 4, 'Fe3+': 5, 'Mo': 5, 'W': 5, 'Ni': 2, 'Fe2+': 4}, 'ALGO': 'FAST', 'LDAU': True, 'IBRION': 2}}</em><a class="headerlink" href="#pymatgen.io.vasp.sets.MITRelaxSet.CONFIG" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.sets.MPHSEBSSet">
<em class="property">class </em><code class="descname">MPHSEBSSet</code><span class="sig-paren">(</span><em>structure</em>, <em>user_incar_settings=None</em>, <em>added_kpoints=None</em>, <em>mode='Uniform'</em>, <em>reciprocal_density=None</em>, <em>kpoints_line_density=20</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#MPHSEBSSet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.MPHSEBSSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#pymatgen.io.vasp.sets.MPHSERelaxSet" title="pymatgen.io.vasp.sets.MPHSERelaxSet"><code class="xref py py-class docutils literal"><span class="pre">pymatgen.io.vasp.sets.MPHSERelaxSet</span></code></a></p>
<p>Implementation of a VaspInputSet for HSE band structure computations.
Remember that HSE band structures must be self-consistent in VASP. A
band structure along symmetry lines for instance needs BOTH a uniform
grid with appropriate weights AND a path along the lines with weight 0.</p>
<p>Thus, the &#8220;Uniform&#8221; mode is just like regular static SCF but allows
adding custom kpoints (e.g., corresponding to known VBM/CBM) to the
uniform grid that have zero weight (e.g., for better gap estimate).</p>
<p>The &#8220;Line&#8221; mode is just like Uniform mode, but additionally adds
k-points along symmetry lines with zero weight.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>structure</strong> (<a class="reference internal" href="pymatgen.core.html#pymatgen.core.structure.Structure" title="pymatgen.core.structure.Structure"><em>Structure</em></a>) &#8211; Structure to compute</li>
<li><strong>user_incar_settings</strong> (<em>dict</em>) &#8211; A dict specifying additional incar
settings</li>
<li><strong>added_kpoints</strong> (<em>list</em>) &#8211; a list of kpoints (list of 3 number list)
added to the run. The k-points are in fractional coordinates</li>
<li><strong>mode</strong> (<em>str</em>) &#8211; &#8220;Line&#8221; - generate k-points along symmetry lines for
bandstructure. &#8220;Uniform&#8221; - generate uniform k-points grid</li>
<li><strong>reciprocal_density</strong> (<em>int</em>) &#8211; k-point density to use for uniform mesh</li>
<li><strong>kpoints_line_density</strong> (<em>int</em>) &#8211; k-point density for high symmetry lines</li>
<li><strong>**kwargs</strong> (<em>dict</em>) &#8211; Any other parameters to pass into DictVaspInputSet</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="classmethod">
<dt id="pymatgen.io.vasp.sets.MPHSEBSSet.from_prev_calc">
<em class="property">classmethod </em><code class="descname">from_prev_calc</code><span class="sig-paren">(</span><em>prev_calc_dir</em>, <em>mode='Uniform'</em>, <em>reciprocal_density=50</em>, <em>copy_chgcar=True</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#MPHSEBSSet.from_prev_calc"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.MPHSEBSSet.from_prev_calc" title="Permalink to this definition">¶</a></dt>
<dd><p>Generate a set of Vasp input files for HSE calculations from a
directory of previous Vasp run. Explicitly adds VBM and CBM of prev.
run to the k-point list of this run.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>prev_calc_dir</strong> (<em>str</em>) &#8211; Directory containing the outputs
(vasprun.xml and OUTCAR) of previous vasp run.</li>
<li><strong>mode</strong> (<em>str</em>) &#8211; Either &#8220;Uniform&#8221; or &#8220;Line&#8221;</li>
<li><strong>reciprocal_density</strong> (<em>int</em>) &#8211; density of k-mesh</li>
<li><strong>copy_chgcar</strong> (<em>bool</em>) &#8211; whether to copy CHGCAR of previous run</li>
<li><strong>**kwargs</strong> &#8211; All kwargs supported by MPHSEBSStaticSet,
other than prev_structure which is determined from the previous
calc dir.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.MPHSEBSSet.kpoints">
<code class="descname">kpoints</code><a class="headerlink" href="#pymatgen.io.vasp.sets.MPHSEBSSet.kpoints" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.sets.MPHSERelaxSet">
<em class="property">class </em><code class="descname">MPHSERelaxSet</code><span class="sig-paren">(</span><em>structure</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#MPHSERelaxSet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.MPHSERelaxSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#pymatgen.io.vasp.sets.DictSet" title="pymatgen.io.vasp.sets.DictSet"><code class="xref py py-class docutils literal"><span class="pre">pymatgen.io.vasp.sets.DictSet</span></code></a></p>
<p>Same as the MPRelaxSet, but with HSE parameters.</p>
<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.MPHSERelaxSet.CONFIG">
<code class="descname">CONFIG</code><em class="property"> = {'KPOINTS': {'reciprocal_density': 50}, 'POTCAR': {'Th': 'Th', 'Rb': 'Rb_sv', 'Ca': 'Ca_sv', 'Tm': 'Tm_3', 'Sc': 'Sc_sv', 'Pm': 'Pm_3', 'Si': 'Si', 'Mn': 'Mn_pv', 'Tl': 'Tl_d', 'Os': 'Os_pv', 'Rh': 'Rh_pv', 'Zn': 'Zn', 'Xe': 'Xe', 'Nb': 'Nb_pv', 'K': 'K_sv', 'Ti': 'Ti_pv', 'O': 'O', 'Pt': 'Pt', 'Er': 'Er_3', 'Bi': 'Bi', 'C': 'C', 'Sb': 'Sb', 'Sr': 'Sr_sv', 'P': 'P', 'Cd': 'Cd', 'Pa': 'Pa', 'U': 'U', 'Pb': 'Pb_d', 'Eu': 'Eu', 'Kr': 'Kr', 'Li': 'Li_sv', 'Ga': 'Ga_d', 'Dy': 'Dy_3', 'Ni': 'Ni_pv', 'Ag': 'Ag', 'Ge': 'Ge_d', 'H': 'H', 'Ce': 'Ce', 'He': 'He', 'Re': 'Re_pv', 'V': 'V_pv', 'In': 'In_d', 'Se': 'Se', 'Ir': 'Ir', 'Cl': 'Cl', 'W': 'W_pv', 'Cu': 'Cu_pv', 'Ho': 'Ho_3', 'Ne': 'Ne', 'Ac': 'Ac', 'Te': 'Te', 'N': 'N', 'Nd': 'Nd_3', 'Ta': 'Ta_pv', 'Gd': 'Gd', 'Fe': 'Fe_pv', 'Pd': 'Pd', 'Np': 'Np', 'As': 'As', 'Co': 'Co', 'Br': 'Br', 'F': 'F', 'La': 'La', 'Be': 'Be_sv', 'Tb': 'Tb_3', 'Na': 'Na_pv', 'Zr': 'Zr_sv', 'Cs': 'Cs_sv', 'Sm': 'Sm_3', 'Tc': 'Tc_pv', 'Al': 'Al', 'B': 'B', 'Ar': 'Ar', 'Hf': 'Hf_pv', 'Ba': 'Ba_sv', 'Mg': 'Mg_pv', 'Ru': 'Ru_pv', 'Pr': 'Pr_3', 'S': 'S', 'Y': 'Y_sv', 'Hg': 'Hg', 'Yb': 'Yb_2', 'Lu': 'Lu_3', 'Sn': 'Sn_d', 'Mo': 'Mo_pv', 'I': 'I', 'Pu': 'Pu', 'Cr': 'Cr_pv', 'Au': 'Au'}, 'INCAR': {'LWAVE': False, 'NELM': 100, 'HFSCREEN': 0.2, 'PREC': 'Accurate', 'EDIFF_PER_ATOM': 5e-05, 'NSW': 99, 'ISPIN': 2, 'LORBIT': 11, 'ENCUT': 520, 'ISMEAR': 0, 'ICHARG': 1, 'LHFCALC': True, 'SIGMA': 0.05, 'LREAL': 'AUTO', 'ISIF': 3, 'PRECFOCK': 'Fast', 'ALGO': 'All', 'MAGMOM': {'V': 5, 'Mo': 5, 'Eu': 10, 'Fe': 5, 'W': 5, 'Mn': 5, 'Cr': 5, 'Mn3+': 4, 'Ce': 5, 'Ni': 5, 'Co': 5, 'Mn4+': 3}, 'IBRION': 2}}</em><a class="headerlink" href="#pymatgen.io.vasp.sets.MPHSERelaxSet.CONFIG" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.sets.MPNonSCFSet">
<em class="property">class </em><code class="descname">MPNonSCFSet</code><span class="sig-paren">(</span><em>structure</em>, <em>prev_incar=None</em>, <em>mode='line'</em>, <em>nedos=601</em>, <em>reciprocal_density=100</em>, <em>sym_prec=0.1</em>, <em>kpoints_line_density=20</em>, <em>optics=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#MPNonSCFSet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.MPNonSCFSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#pymatgen.io.vasp.sets.MPRelaxSet" title="pymatgen.io.vasp.sets.MPRelaxSet"><code class="xref py py-class docutils literal"><span class="pre">pymatgen.io.vasp.sets.MPRelaxSet</span></code></a></p>
<p>Init a MPNonSCFSet. Typically, you would use the classmethod
from_prev_calc to initialize from a previous SCF run.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>structure</strong> (<a class="reference internal" href="pymatgen.core.html#pymatgen.core.structure.Structure" title="pymatgen.core.structure.Structure"><em>Structure</em></a>) &#8211; Structure to compute</li>
<li><strong>prev_incar</strong> (<a class="reference internal" href="#pymatgen.io.vasp.inputs.Incar" title="pymatgen.io.vasp.inputs.Incar"><em>Incar</em></a>) &#8211; Incar file from previous run.</li>
<li><strong>mode</strong> (<em>str</em>) &#8211; Line or Uniform mode supported.</li>
<li><strong>nedos</strong> (<em>int</em>) &#8211; nedos parameter. Default to 601.</li>
<li><strong>reciprocal_density</strong> (<em>int</em>) &#8211; density of k-mesh by reciprocal
volume (defaults to 100)</li>
<li><strong>sym_prec</strong> (<em>float</em>) &#8211; Symmetry precision (for Uniform mode).</li>
<li><strong>kpoints_line_density</strong> (<em>int</em>) &#8211; Line density for Line mode.</li>
<li><strong>optics</strong> (<em>bool</em>) &#8211; whether to add dielectric function</li>
<li><strong>**kwargs</strong> &#8211; kwargs supported by MPVaspInputSet.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="classmethod">
<dt id="pymatgen.io.vasp.sets.MPNonSCFSet.from_prev_calc">
<em class="property">classmethod </em><code class="descname">from_prev_calc</code><span class="sig-paren">(</span><em>prev_calc_dir</em>, <em>copy_chgcar=True</em>, <em>nbands_factor=1.2</em>, <em>standardize=False</em>, <em>sym_prec=0.1</em>, <em>international_monoclinic=True</em>, <em>reciprocal_density=100</em>, <em>kpoints_line_density=20</em>, <em>small_gap_multiply=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#MPNonSCFSet.from_prev_calc"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.MPNonSCFSet.from_prev_calc" title="Permalink to this definition">¶</a></dt>
<dd><p>Generate a set of Vasp input files for NonSCF calculations from a
directory of previous static Vasp run.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>prev_calc_dir</strong> (<em>str</em>) &#8211; The directory contains the outputs(
vasprun.xml and OUTCAR) of previous vasp run.</li>
<li><strong>copy_chgcar</strong> &#8211; Whether to copy the old CHGCAR. Defaults to True.</li>
<li><strong>nbands_factor</strong> (<em>float</em>) &#8211; Multiplicative factor for NBANDS. Choose a
higher number if you are doing an LOPTICS calculation.</li>
<li><strong>standardize</strong> (<em>float</em>) &#8211; Whether to standardize to a primitive
standard cell. Defaults to False.</li>
<li><strong>sym_prec</strong> (<em>float</em>) &#8211; Tolerance for symmetry finding. If not 0,
the final structure from the previous run will be symmetrized
to get a primitive standard cell. Set to 0 if you don&#8217;t want
that.</li>
<li><strong>international_monoclinic</strong> (<em>bool</em>) &#8211; Whether to use international
convention (vs Curtarolo) for monoclinic. Defaults True.</li>
<li><strong>reciprocal_density</strong> (<em>int</em>) &#8211; density of k-mesh by reciprocal
volume in uniform mode (defaults to 100)</li>
<li><strong>kpoints_line_density</strong> (<em>int</em>) &#8211; density of k-mesh in line mode
(defaults to 20)</li>
<li><strong>small_gap_multiply</strong> (<em>[float, float]</em>) &#8211; If the gap is less than
1st index, multiply the default reciprocal_density by the 2nd
index.</li>
<li><strong>**kwargs</strong> &#8211; All kwargs supported by MPNonSCFSet,
other than structure, prev_incar and prev_chgcar which
are determined from the prev_calc_dir.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.MPNonSCFSet.incar">
<code class="descname">incar</code><a class="headerlink" href="#pymatgen.io.vasp.sets.MPNonSCFSet.incar" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.MPNonSCFSet.kpoints">
<code class="descname">kpoints</code><a class="headerlink" href="#pymatgen.io.vasp.sets.MPNonSCFSet.kpoints" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.sets.MPRelaxSet">
<em class="property">class </em><code class="descname">MPRelaxSet</code><span class="sig-paren">(</span><em>structure</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#MPRelaxSet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.MPRelaxSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#pymatgen.io.vasp.sets.DictSet" title="pymatgen.io.vasp.sets.DictSet"><code class="xref py py-class docutils literal"><span class="pre">pymatgen.io.vasp.sets.DictSet</span></code></a></p>
<p>Implementation of VaspInputSet utilizing parameters in the public
Materials Project. Typically, the pseudopotentials chosen contain more
electrons than the MIT parameters, and the k-point grid is ~50% more dense.
The LDAUU parameters are also different due to the different psps used,
which result in different fitted values.</p>
<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.MPRelaxSet.CONFIG">
<code class="descname">CONFIG</code><em class="property"> = {'KPOINTS': {'reciprocal_density': 50}, 'POTCAR': {'Th': 'Th', 'Rb': 'Rb_sv', 'Ca': 'Ca_sv', 'Tm': 'Tm_3', 'Sc': 'Sc_sv', 'Pm': 'Pm_3', 'Si': 'Si', 'Mn': 'Mn_pv', 'Tl': 'Tl_d', 'Os': 'Os_pv', 'Rh': 'Rh_pv', 'Zn': 'Zn', 'Xe': 'Xe', 'Nb': 'Nb_pv', 'K': 'K_sv', 'Ti': 'Ti_pv', 'O': 'O', 'Pt': 'Pt', 'Er': 'Er_3', 'Bi': 'Bi', 'C': 'C', 'Sb': 'Sb', 'Sr': 'Sr_sv', 'P': 'P', 'Cd': 'Cd', 'Pa': 'Pa', 'U': 'U', 'Pb': 'Pb_d', 'Eu': 'Eu', 'Kr': 'Kr', 'Li': 'Li_sv', 'Ga': 'Ga_d', 'Dy': 'Dy_3', 'Ni': 'Ni_pv', 'Ag': 'Ag', 'Ge': 'Ge_d', 'H': 'H', 'Ce': 'Ce', 'He': 'He', 'Re': 'Re_pv', 'V': 'V_pv', 'In': 'In_d', 'Se': 'Se', 'Ir': 'Ir', 'Cl': 'Cl', 'W': 'W_pv', 'Cu': 'Cu_pv', 'Ho': 'Ho_3', 'Ne': 'Ne', 'Ac': 'Ac', 'Te': 'Te', 'N': 'N', 'Nd': 'Nd_3', 'Ta': 'Ta_pv', 'Gd': 'Gd', 'Fe': 'Fe_pv', 'Pd': 'Pd', 'Np': 'Np', 'As': 'As', 'Co': 'Co', 'Br': 'Br', 'F': 'F', 'La': 'La', 'Be': 'Be_sv', 'Tb': 'Tb_3', 'Na': 'Na_pv', 'Zr': 'Zr_sv', 'Cs': 'Cs_sv', 'Sm': 'Sm_3', 'Tc': 'Tc_pv', 'Al': 'Al', 'B': 'B', 'Ar': 'Ar', 'Hf': 'Hf_pv', 'Ba': 'Ba_sv', 'Mg': 'Mg_pv', 'Ru': 'Ru_pv', 'Pr': 'Pr_3', 'S': 'S', 'Y': 'Y_sv', 'Hg': 'Hg', 'Yb': 'Yb_2', 'Lu': 'Lu_3', 'Sn': 'Sn_d', 'Mo': 'Mo_pv', 'I': 'I', 'Pu': 'Pu', 'Cr': 'Cr_pv', 'Au': 'Au'}, 'INCAR': {'LWAVE': False, 'LDAUL': {'O': {'V': 2, 'Mo': 2, 'Fe': 2, 'W': 2, 'Mn': 2, 'Cr': 2, 'Co': 2, 'Ni': 2}, 'F': {'V': 2, 'Mo': 2, 'Fe': 2, 'W': 2, 'Mn': 2, 'Cr': 2, 'Co': 2, 'Ni': 2}}, 'NELM': 100, 'PREC': 'Accurate', 'LDAUTYPE': 2, 'EDIFF_PER_ATOM': 5e-05, 'ENCUT': 520, 'NSW': 99, 'LDAUPRINT': 1, 'ISIF': 3, 'LORBIT': 11, 'LDAUU': {'O': {'V': 3.25, 'Mo': 4.38, 'Fe': 5.3, 'W': 6.2, 'Mn': 3.9, 'Cr': 3.7, 'Co': 3.32, 'Ni': 6.2}, 'F': {'V': 3.25, 'Mo': 4.38, 'Fe': 5.3, 'W': 6.2, 'Mn': 3.9, 'Cr': 3.7, 'Co': 3.32, 'Ni': 6.2}}, 'ISMEAR': -5, 'ICHARG': 1, 'LDAUJ': {'O': {'V': 0, 'Mo': 0, 'Fe': 0, 'W': 0, 'Mn': 0, 'Cr': 0, 'Co': 0, 'Ni': 0}, 'F': {'V': 0, 'Mo': 0, 'Fe': 0, 'W': 0, 'Mn': 0, 'Cr': 0, 'Co': 0, 'Ni': 0}}, 'SIGMA': 0.05, 'LREAL': 'AUTO', 'ISPIN': 2, 'MAGMOM': {'V': 5, 'Mo': 5, 'Co3+': 0.6, 'Eu': 10, 'Fe': 5, 'Co4+': 1, 'W': 5, 'Mn': 5, 'Ni': 5, 'Mn3+': 4, 'Co': 5, 'Cr': 5, 'Ce': 5, 'Mn4+': 3}, 'ALGO': 'FAST', 'LDAU': True, 'IBRION': 2}}</em><a class="headerlink" href="#pymatgen.io.vasp.sets.MPRelaxSet.CONFIG" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.sets.MPSOCSet">
<em class="property">class </em><code class="descname">MPSOCSet</code><span class="sig-paren">(</span><em>structure</em>, <em>saxis=(0</em>, <em>0</em>, <em>1)</em>, <em>prev_incar=None</em>, <em>reciprocal_density=100</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#MPSOCSet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.MPSOCSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#pymatgen.io.vasp.sets.MPStaticSet" title="pymatgen.io.vasp.sets.MPStaticSet"><code class="xref py py-class docutils literal"><span class="pre">pymatgen.io.vasp.sets.MPStaticSet</span></code></a></p>
<p>Init a MPSOCSet.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>structure</strong> (<a class="reference internal" href="pymatgen.core.html#pymatgen.core.structure.Structure" title="pymatgen.core.structure.Structure"><em>Structure</em></a>) &#8211; the structure must have the &#8216;magmom&#8217; site
property and each magnetic moment value must have 3
components. eg:- magmom = [[0,0,2], ...]</li>
<li><strong>saxis</strong> (<em>tuple</em>) &#8211; magnetic moment orientation</li>
<li><strong>prev_incar</strong> (<a class="reference internal" href="#pymatgen.io.vasp.inputs.Incar" title="pymatgen.io.vasp.inputs.Incar"><em>Incar</em></a>) &#8211; Incar file from previous run.</li>
<li><strong>reciprocal_density</strong> (<em>int</em>) &#8211; density of k-mesh by reciprocal
volume (defaults to 100)</li>
<li><strong>**kwargs</strong> &#8211; kwargs supported by MPVaspInputSet.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="classmethod">
<dt id="pymatgen.io.vasp.sets.MPSOCSet.from_prev_calc">
<em class="property">classmethod </em><code class="descname">from_prev_calc</code><span class="sig-paren">(</span><em>prev_calc_dir</em>, <em>copy_chgcar=True</em>, <em>nbands_factor=1.2</em>, <em>standardize=False</em>, <em>sym_prec=0.1</em>, <em>international_monoclinic=True</em>, <em>reciprocal_density=100</em>, <em>small_gap_multiply=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#MPSOCSet.from_prev_calc"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.MPSOCSet.from_prev_calc" title="Permalink to this definition">¶</a></dt>
<dd><p>Generate a set of Vasp input files for SOC calculations from a
directory of previous static Vasp run. SOC calc requires all 3
components for MAGMOM for each atom in the structure.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>prev_calc_dir</strong> (<em>str</em>) &#8211; The directory contains the outputs(
vasprun.xml and OUTCAR) of previous vasp run.</li>
<li><strong>copy_chgcar</strong> &#8211; Whether to copy the old CHGCAR. Defaults to True.</li>
<li><strong>nbands_factor</strong> (<em>float</em>) &#8211; Multiplicative factor for NBANDS. Choose a
higher number if you are doing an LOPTICS calculation.</li>
<li><strong>standardize</strong> (<em>float</em>) &#8211; Whether to standardize to a primitive
standard cell. Defaults to False.</li>
<li><strong>sym_prec</strong> (<em>float</em>) &#8211; Tolerance for symmetry finding. If not 0,
the final structure from the previous run will be symmetrized
to get a primitive standard cell. Set to 0 if you don&#8217;t want
that.</li>
<li><strong>international_monoclinic</strong> (<em>bool</em>) &#8211; Whether to use international
convention (vs Curtarolo) for monoclinic. Defaults True.</li>
<li><strong>reciprocal_density</strong> (<em>int</em>) &#8211; density of k-mesh by reciprocal
volume (defaults to 100)</li>
<li><strong>small_gap_multiply</strong> (<em>[float, float]</em>) &#8211; If the gap is less than
1st index, multiply the default reciprocal_density by the 2nd
index.</li>
<li><strong>**kwargs</strong> &#8211; All kwargs supported by MPSOCSet,
other than structure, prev_incar and prev_chgcar which
are determined from the prev_calc_dir.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.MPSOCSet.incar">
<code class="descname">incar</code><a class="headerlink" href="#pymatgen.io.vasp.sets.MPSOCSet.incar" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.sets.MPStaticSet">
<em class="property">class </em><code class="descname">MPStaticSet</code><span class="sig-paren">(</span><em>structure</em>, <em>prev_incar=None</em>, <em>prev_kpoints=None</em>, <em>lepsilon=False</em>, <em>reciprocal_density=100</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#MPStaticSet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.MPStaticSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#pymatgen.io.vasp.sets.MPRelaxSet" title="pymatgen.io.vasp.sets.MPRelaxSet"><code class="xref py py-class docutils literal"><span class="pre">pymatgen.io.vasp.sets.MPRelaxSet</span></code></a></p>
<p>Run a static calculation.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>structure</strong> (<a class="reference internal" href="pymatgen.core.html#pymatgen.core.structure.Structure" title="pymatgen.core.structure.Structure"><em>Structure</em></a>) &#8211; Structure from previous run.</li>
<li><strong>prev_incar</strong> (<a class="reference internal" href="#pymatgen.io.vasp.inputs.Incar" title="pymatgen.io.vasp.inputs.Incar"><em>Incar</em></a>) &#8211; Incar file from previous run.</li>
<li><strong>prev_kpoints</strong> (<a class="reference internal" href="#pymatgen.io.vasp.inputs.Kpoints" title="pymatgen.io.vasp.inputs.Kpoints"><em>Kpoints</em></a>) &#8211; Kpoints from previous run.</li>
<li><strong>lepsilon</strong> (<em>bool</em>) &#8211; Whether to add static dielectric calculation</li>
<li><strong>reciprocal_density</strong> (<em>int</em>) &#8211; For static calculations,
we usually set the reciprocal density by voluyme. This is a
convenience arg to change that, rather than using
user_kpoints_settings. Defaults to 100.</li>
<li><strong>**kwargs</strong> &#8211; kwargs supported by MPRelaxSet.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="classmethod">
<dt id="pymatgen.io.vasp.sets.MPStaticSet.from_prev_calc">
<em class="property">classmethod </em><code class="descname">from_prev_calc</code><span class="sig-paren">(</span><em>prev_calc_dir</em>, <em>standardize=False</em>, <em>sym_prec=0.1</em>, <em>international_monoclinic=True</em>, <em>reciprocal_density=100</em>, <em>small_gap_multiply=None</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#MPStaticSet.from_prev_calc"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.MPStaticSet.from_prev_calc" title="Permalink to this definition">¶</a></dt>
<dd><p>Generate a set of Vasp input files for static calculations from a
directory of previous Vasp run.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>prev_calc_dir</strong> (<em>str</em>) &#8211; Directory containing the outputs(
vasprun.xml and OUTCAR) of previous vasp run.</li>
<li><strong>standardize</strong> (<em>float</em>) &#8211; Whether to standardize to a primitive
standard cell. Defaults to False.</li>
<li><strong>sym_prec</strong> (<em>float</em>) &#8211; Tolerance for symmetry finding. If not 0,
the final structure from the previous run will be symmetrized
to get a primitive standard cell. Set to 0 if you don&#8217;t want
that.</li>
<li><strong>international_monoclinic</strong> (<em>bool</em>) &#8211; Whether to use international
convention (vs Curtarolo) for monoclinic. Defaults True.</li>
<li><strong>reciprocal_density</strong> (<em>int</em>) &#8211; density of k-mesh by reciprocal
volume (defaults to 100)</li>
<li><strong>small_gap_multiply</strong> (<em>[float, float]</em>) &#8211; If the gap is less than
1st index, multiply the default reciprocal_density by the 2nd
index.</li>
<li><strong>**kwargs</strong> &#8211; All kwargs supported by MPStaticSet,
other than prev_incar and prev_structure and prev_kpoints which
are determined from the prev_calc_dir.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.MPStaticSet.incar">
<code class="descname">incar</code><a class="headerlink" href="#pymatgen.io.vasp.sets.MPStaticSet.incar" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.MPStaticSet.kpoints">
<code class="descname">kpoints</code><a class="headerlink" href="#pymatgen.io.vasp.sets.MPStaticSet.kpoints" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.sets.MVLElasticSet">
<em class="property">class </em><code class="descname">MVLElasticSet</code><span class="sig-paren">(</span><em>structure</em>, <em>potim=0.015</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#MVLElasticSet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.MVLElasticSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#pymatgen.io.vasp.sets.MPRelaxSet" title="pymatgen.io.vasp.sets.MPRelaxSet"><code class="xref py py-class docutils literal"><span class="pre">pymatgen.io.vasp.sets.MPRelaxSet</span></code></a></p>
<p>MVL denotes VASP input sets that are implemented by the Materials Virtual
Lab (<a class="reference external" href="http://www.materialsvirtuallab.org">http://www.materialsvirtuallab.org</a>) for various research.</p>
<p>This input set is used to calculate elastic constants in VASP. It is used
in the following work:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span>Z. Deng, Z. Wang, I.-H. Chu, J. Luo, S. P. Ong.
“Elastic Properties of Alkali Superionic Conductor Electrolytes
from First Principles Calculations”, J. Electrochem. Soc.
2016, 163(2), A67-A74. doi: 10.1149/2.0061602jes
</pre></div>
</div>
<p>To read the elastic constants, you may use the Outcar class which parses the
elastic constants.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>scale</strong> (<em>float</em>) &#8211; POTIM parameter. The default of 0.015 is usually fine,
but some structures may require a smaller step.</li>
<li><strong>user_incar_settings</strong> (<em>dict</em>) &#8211; A dict specifying additional incar
settings.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.sets.MVLSlabSet">
<em class="property">class </em><code class="descname">MVLSlabSet</code><span class="sig-paren">(</span><em>structure</em>, <em>gpu=False</em>, <em>k_product=50</em>, <em>bulk=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#MVLSlabSet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.MVLSlabSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#pymatgen.io.vasp.sets.MPRelaxSet" title="pymatgen.io.vasp.sets.MPRelaxSet"><code class="xref py py-class docutils literal"><span class="pre">pymatgen.io.vasp.sets.MPRelaxSet</span></code></a></p>
<p>Class for writing a set of slab vasp runs,
including both slabs (along the c direction) and orient unit cells (bulk),
to ensure the same KPOINTS, POTCAR and INCAR criterion.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>k_product</strong> &#8211; default to 50, kpoint number * length for a &amp; b directions,
also for c direction in bulk calculations</li>
<li><strong>bulk</strong> (<em>bool</em>) &#8211; Set to True for bulk calculation. Defaults to False.</li>
<li><strong>**kwargs</strong> &#8211; Other kwargs supported by <a class="reference internal" href="#pymatgen.io.vasp.sets.DictSet" title="pymatgen.io.vasp.sets.DictSet"><code class="xref py py-class docutils literal"><span class="pre">DictSet</span></code></a>.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.MVLSlabSet.incar">
<code class="descname">incar</code><a class="headerlink" href="#pymatgen.io.vasp.sets.MVLSlabSet.incar" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.MVLSlabSet.kpoints">
<code class="descname">kpoints</code><a class="headerlink" href="#pymatgen.io.vasp.sets.MVLSlabSet.kpoints" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>k_product, default to 50, is kpoint number * length for a &amp; b</dt>
<dd>directions, also for c direction in bulk calculations</dd>
</dl>
<p>Automatic mesh &amp; Gamma is the default setting.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="pymatgen.io.vasp.sets.VaspInputSet">
<em class="property">class </em><code class="descname">VaspInputSet</code><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#VaspInputSet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.VaspInputSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal"><span class="pre">monty.json.MSONable</span></code></p>
<p>Base class representing a set of Vasp input parameters with a structure
supplied as init parameters. Typically, you should not inherit from this
class. Start from DictSet or MPRelaxSet or MITRelaxSet.</p>
<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.VaspInputSet.all_input">
<code class="descname">all_input</code><a class="headerlink" href="#pymatgen.io.vasp.sets.VaspInputSet.all_input" title="Permalink to this definition">¶</a></dt>
<dd><p><em>Returns all input files as a dict of {filename</em> &#8211; vasp object}</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">object}, e.g., {&#8216;INCAR&#8217;: Incar object, ...}</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">dict of {filename</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.sets.VaspInputSet.as_dict">
<code class="descname">as_dict</code><span class="sig-paren">(</span><em>verbosity=2</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#VaspInputSet.as_dict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.VaspInputSet.as_dict" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.VaspInputSet.incar">
<code class="descname">incar</code><a class="headerlink" href="#pymatgen.io.vasp.sets.VaspInputSet.incar" title="Permalink to this definition">¶</a></dt>
<dd><p>Incar object</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.VaspInputSet.kpoints">
<code class="descname">kpoints</code><a class="headerlink" href="#pymatgen.io.vasp.sets.VaspInputSet.kpoints" title="Permalink to this definition">¶</a></dt>
<dd><p>Kpoints object</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.VaspInputSet.poscar">
<code class="descname">poscar</code><a class="headerlink" href="#pymatgen.io.vasp.sets.VaspInputSet.poscar" title="Permalink to this definition">¶</a></dt>
<dd><p>Poscar object</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.VaspInputSet.potcar">
<code class="descname">potcar</code><a class="headerlink" href="#pymatgen.io.vasp.sets.VaspInputSet.potcar" title="Permalink to this definition">¶</a></dt>
<dd><p>Potcar object.</p>
</dd></dl>

<dl class="attribute">
<dt id="pymatgen.io.vasp.sets.VaspInputSet.potcar_symbols">
<code class="descname">potcar_symbols</code><a class="headerlink" href="#pymatgen.io.vasp.sets.VaspInputSet.potcar_symbols" title="Permalink to this definition">¶</a></dt>
<dd><p>List of POTCAR symbols.</p>
</dd></dl>

<dl class="method">
<dt id="pymatgen.io.vasp.sets.VaspInputSet.write_input">
<code class="descname">write_input</code><span class="sig-paren">(</span><em>output_dir</em>, <em>make_dir_if_not_present=True</em>, <em>include_cif=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#VaspInputSet.write_input"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.VaspInputSet.write_input" title="Permalink to this definition">¶</a></dt>
<dd><p>Writes a set of VASP input to a directory.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>output_dir</strong> (<em>str</em>) &#8211; Directory to output the VASP input files</li>
<li><strong>make_dir_if_not_present</strong> (<em>bool</em>) &#8211; Set to True if you want the
directory (and the whole path) to be created if it is not
present.</li>
<li><strong>include_cif</strong> (<em>bool</em>) &#8211; Whether to write a CIF file in the output
directory for easier opening by VESTA.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="pymatgen.io.vasp.sets.batch_write_input">
<code class="descname">batch_write_input</code><span class="sig-paren">(</span><em>structures</em>, <em>vasp_input_set=&lt;class 'pymatgen.io.vasp.sets.MPRelaxSet'&gt;</em>, <em>output_dir='.'</em>, <em>make_dir_if_not_present=True</em>, <em>subfolder=None</em>, <em>sanitize=False</em>, <em>include_cif=False</em>, <em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#batch_write_input"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.batch_write_input" title="Permalink to this definition">¶</a></dt>
<dd><p>Batch write vasp input for a sequence of structures to
output_dir, following the format output_dir/{group}/{formula}_{number}.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>structures</strong> (<em>[Structure]</em>) &#8211; Sequence of Structures.</li>
<li><strong>vasp_input_set</strong> (<a class="reference internal" href="#pymatgen.io.vasp.sets.VaspInputSet" title="pymatgen.io.vasp.sets.VaspInputSet"><em>VaspInputSet</em></a>) &#8211; VaspInputSet class that creates
vasp input files from structures. Note that a class should be
supplied. Defaults to MPRelaxSet.</li>
<li><strong>output_dir</strong> (<em>str</em>) &#8211; Directory to output files. Defaults to current
directory &#8221;.&#8221;.</li>
<li><strong>make_dir_if_not_present</strong> (<em>bool</em>) &#8211; Create the directory if not present.
Defaults to True.</li>
<li><strong>subfolder</strong> (<em>callable</em>) &#8211; Function to create subdirectory name from
structure. Defaults to simply &#8220;formula_count&#8221;.</li>
<li><strong>sanitize</strong> (<em>bool</em>) &#8211; Boolean indicating whether to sanitize the
structure before writing the VASP input files. Sanitized output
are generally easier for viewing and certain forms of analysis.
Defaults to False.</li>
<li><strong>include_cif</strong> (<em>bool</em>) &#8211; Whether to output a CIF as well. CIF files are
generally better supported in visualization programs.</li>
<li><strong>**kwargs</strong> &#8211; Additional kwargs are passed to the vasp_input_set class in
addition to structure.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pymatgen.io.vasp.sets.get_structure_from_prev_run">
<code class="descname">get_structure_from_prev_run</code><span class="sig-paren">(</span><em>vasprun</em>, <em>outcar=None</em>, <em>sym_prec=0.1</em>, <em>international_monoclinic=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#get_structure_from_prev_run"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.get_structure_from_prev_run" title="Permalink to this definition">¶</a></dt>
<dd><p>Process structure from previous run.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>vasprun</strong> (<a class="reference internal" href="#pymatgen.io.vasp.outputs.Vasprun" title="pymatgen.io.vasp.outputs.Vasprun"><em>Vasprun</em></a>) &#8211; Vasprun that contains the final structure
from previous run.</li>
<li><strong>outcar</strong> (<a class="reference internal" href="#pymatgen.io.vasp.outputs.Outcar" title="pymatgen.io.vasp.outputs.Outcar"><em>Outcar</em></a>) &#8211; Outcar that contains the magnetization info from
previous run.</li>
<li><strong>sym_prec</strong> (<em>float</em>) &#8211; Tolerance for symmetry finding for standardization. If
no standardization is desired, set to 0 or a False.</li>
<li><strong>international_monoclinic</strong> (<em>bool</em>) &#8211; Whether to use international
convention (vs Curtarolo) for monoclinic. Defaults True.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">Returns the magmom-decorated structure that can be passed to get
Vasp input files, e.g. get_kpoints.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="pymatgen.io.vasp.sets.get_vasprun_outcar">
<code class="descname">get_vasprun_outcar</code><span class="sig-paren">(</span><em>path</em>, <em>parse_dos=True</em>, <em>parse_eigen=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/pymatgen/io/vasp/sets.html#get_vasprun_outcar"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#pymatgen.io.vasp.sets.get_vasprun_outcar" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</div>
<div class="section" id="module-pymatgen.io.vasp">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-pymatgen.io.vasp" title="Permalink to this headline">¶</a></h2>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">pymatgen.io.vasp package</a><ul>
<li><a class="reference internal" href="#subpackages">Subpackages</a></li>
<li><a class="reference internal" href="#submodules">Submodules</a></li>
<li><a class="reference internal" href="#pymatgen-io-vasp-gwvaspinputsets-module">pymatgen.io.vasp.GWvaspinputsets module</a></li>
<li><a class="reference internal" href="#module-pymatgen.io.vasp.inputs">pymatgen.io.vasp.inputs module</a></li>
<li><a class="reference internal" href="#module-pymatgen.io.vasp.outputs">pymatgen.io.vasp.outputs module</a></li>
<li><a class="reference internal" href="#module-pymatgen.io.vasp.sets">pymatgen.io.vasp.sets module</a></li>
<li><a class="reference internal" href="#module-pymatgen.io.vasp">Module contents</a></li>
</ul>
</li>
</ul>

  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/pymatgen.io.vasp.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">pymatgen 4.4.9 documentation</a> &#187;</li> 
      </ul>
    </div>

    <div class="footer" role="contentinfo">
        &#169; Copyright 2011, Pymatgen Development Team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.4.8.
    </div>
<div class="footer">This page uses <a href="http://analytics.google.com/">
Google Analytics</a> to collect statistics. You can disable it by blocking
the JavaScript coming from www.google-analytics.com.
<script type="text/javascript">
  (function() {
    var ga = document.createElement('script');
    ga.src = ('https:' == document.location.protocol ?
              'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    ga.setAttribute('async', 'true');
    document.documentElement.firstChild.appendChild(ga);
  })();
</script>
</div>

  </body>
</html>